<!DOCTYPE html>
<html>
    <head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="chrome=1">
        <title>Galaxy Training!</title>
        <link media="all" href="https://netdna.bootstrapcdn.com/bootstrap/3.2.0/css/bootstrap.min.css" rel="stylesheet">
        <link rel="stylesheet" href="https://galaxyproject.github.io/training-material//assets/css/landing_page.css">
        <link rel="stylesheet" href="https://galaxyproject.github.io/training-material//assets/css/tutorial.css">
        <link rel="stylesheet" href="https://galaxyproject.github.io/training-material//assets/css/font-awesome.css">
        <link rel="stylesheet" href="https://galaxyproject.github.io/training-material//assets/css/syntax_highlighting.css">
        <meta name="viewport" content="width=device-width">
    </head>
    <body>
        <div class="wrapper">
            


<header>
    <nav class="navbar navbar-default navbar-fixed-top" role="navigation">
        <div class="container-fluid">
          <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#navbar">
              <span class="sr-only">Toggle navigation</span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
            </button>
            <a id="site-title" class="navbar-brand" href="https://wiki.galaxyproject.org/Teach/GTN">
              <img class="pull-left img-responsive" src="https://galaxyproject.github.io/training-material///assets/images/GTN.png">
            </a>
            <a id="site-title" class="navbar-brand name-holder" href="https://galaxyproject.github.io/training-material//topics/proteomics/tutorials/ntails/tutorial.html">
              ntails
            </a>
          </div>

          <div class="collapse navbar-collapse" id="navbar">

            <ul id="main-menu" class="nav navbar-nav navbar-right">
              <li><a href="https://galaxyproject.github.io/training-material/" title="Back to front page"><i class="fa fa-home"></i> Home</a></li>
              <li><a href="https://galaxyproject.github.io/training-material//topics/proteomics" title="Go back to list of tutorials"><i class="fa fa-file-text-o"></i> Proteomics</a></li>

              
                <li><a href="https://galaxyproject.github.io/training-material//topics/proteomics/slides" title="Slides for this tutorial"><i class="fa fa-graduation-cap"></i> Introduction slides</a></li>
              
              <!--<li><a href="https://galaxyproject.github.io/training-material//proteomics/slides/ntails/"><i class="fa fa-file-powerpoint-o"></i> Associated slides</a></li>-->

              
                <li><a href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/docker" title="Docker image for this tutorial"><i class="fa fa-ship"></i> Docker image</a></li>
              

              

              
                <li><a href="url%20to%20Zenodo%20with%20input%20data" title="Links to data"><i class="fa fa-folder-open"></i> Input Dataset</a></li>
              

              
                <li><a href="https://galaxyproject.github.io/training-material//topics/proteomics#references" title="References"><i class="fa fa-book"></i> Literature</a></li>
              

            </ul>
          </div>
        </div>
    </nav>
</header>

<section class="tutorial">
    <h1>ntails</h1>

    <blockquote class="overview">
        <h3>Overview</h3>

        <p class="bold"> <img class="emoji" title=":grey_question:" alt=":grey_question:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/2754.png" height="20" width="20" align="absmiddle"> Questions</p>
        <ul>
        
          <li>First question addressed during the tutorial</li>
        
          <li>Second question addressed during the tutorial</li>
        
        </ul>

        <p class="bold"> <img class="emoji" title=":dart:" alt=":dart:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f3af.png" height="20" width="20" align="absmiddle"> Objectives</p>
        <ul>
        
          <li>First learning objectives of the tutorial</li>
        
          <li>Second learning objectives of the tutorial</li>
        
        </ul>

        
        <p class="bold"> <img class="emoji" title=":heavy_check_mark:" alt=":heavy_check_mark:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/2714.png" height="20" width="20" align="absmiddle"> Requirements</p>
        <ul>
          
            <li>
                
                <a href="https://galaxyproject.github.io/training-material//topics//introduction/">Galaxy introduction</a>
                
            </li>
          
          
        </ul>
        

        <p><span class="bold"> <img class="emoji" title=":hourglass:" alt=":hourglass:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/231b.png" height="20" width="20" align="absmiddle"> Time estimation:</span> 1h/1d</p>

    </blockquote>

<p>N-Tails is a special Proteomics technique to analyze peptide abundancy changes of protein N-termini. Prior to the MS measurement, N-Tails enriches unmodified, as well as acetylated N-termini. Both common and “unusual” N-termini are identified, where “unusual” means that the protein N-terminus was changes. 
This is best explained by an example: directly after translation, a protein has exactly one N-terminus. When a protease is cutting the protein in half, each half has its own N-terminus. While the N-terminus of the first half protein is exactly the same as the one of the full protein precursor (“native N-terminus”), the N-terminus of the second half is different (“neo-N-terminus”) and depends on the amino acid sequence where the protein was cut. 
The N-Tails technique includes the use of heavy isotope dimethyl labelling.</p>

<p>The figure below illustrates the mechanism of N-Tails. It was originally published by Stefan Tholen (doctoral thesis, not available online). Further reading on N-Tails and other N-terminal techniques, see <a href="https://dx.doi.org/10.1007/978-3-7091-0885-7_5">Tholen et al., Springer Vienna, 2013</a>.</p>

<figure id="figure-1"><img src="../../images/wf_ntails_technique.png" alt="N-Tails technique" title="N-Tails technique"><figcaption><span class="figcaption-prefix">Figure 1:</span> N-Tails technique</figcaption></figure>

<p>The N-Tails technique was originally designed to research protease biology and has most often been used in this field. It was originally published in <a href="https://www.ncbi.nlm.nih.gov/pubmed/20208520">Kleifeld et al., Nat. Biotechnol., 2010</a>.</p>

<blockquote>
  <h3 id="nut_and_bolt-comment-interpretation-of-n-tails-results">
<img class="emoji" title=":nut_and_bolt:" alt=":nut_and_bolt:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f529.png" height="20" width="20" align="absmiddle"> Comment: Interpretation of N-Tails results</h3>

  <p class="comment">Be careful not to overinterprete the results of N-Tails experiments. While the technique <strong>is</strong> fit to identify direct protease substrates, it does not discriminate direct from indirect (“downstream”) effects. Thus, most of the identified N-termini will <strong>not</strong> be direct protease substrates, even if their change in protein abundance is statistically significant.
This warning is not specific for the N-Tails technique, but applies as well to other proteomic N-terminal screening techniques (e.g. COFRADIC).</p>
</blockquote>

<p>This workflow was originally built in the OpenMS framework “TOPPAS” and published in <a href="https://www.ncbi.nlm.nih.gov/pubmed/27087653">Lai, Weisser et al., MCP, 2016</a>. It was converted to OpenMS v2.1, rebuild for the Galaxy framework and tested on the original dataset by Melanie Föll. It was designed for data analysis of a three samples combined in one MS run, a technique based on dimethyl stable isotope labeling (SIL). For more information on SIL, consult <a href="../../tutorials/proteinQuant_SIL.md">this tutorial</a>. The original data were generated using pre-fractionation. Thus, peptides of one biological experiment are measured in multiple consecutive MS runs (one run per fraction).</p>

<p>The figure below gives an overview of the used Galaxy nodes. For further description of the workflow, please consider the <a href="https://www.ncbi.nlm.nih.gov/pubmed/27087653">original publication</a>.</p>

<figure id="figure-2"><img src="../../images/wf_ntails_tripleDimethylLabel.png" alt="N-Tails Galaxy Workflow" title="N-Tails Galaxy Workflow"><figcaption><span class="figcaption-prefix">Figure 2:</span> N-Tails Galaxy Workflow</figcaption></figure>

<h2 id="input">Input</h2>

<p>The workflow needs three input files:</p>

<p>1) A collection of mzML files (multiple fractions of the same experiment).
  2) A FASTA protein database for the organism of interest. For more information on protein databases, consult <a href="../../tutorials/database_handling.md">this tutorial</a></p>

<h2 id="customizing-the-workflow">Customizing the Workflow</h2>

<ul>
  <li>
<em>Running the workflow on a non-prefractionated sample:</em> Simply use only one file as an input.</li>
  <li>
<em>Running the workflow on a double dimethyl labeling (only light and heavy labeling):</em> remove the third <code class="highlighter-rouge">MSGFPlusAdapter</code> <img class="emoji" title=":wrench:" alt=":wrench:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f527.png" height="20" width="20" align="absmiddle"> and the following <code class="highlighter-rouge">PeptideIndexer</code> <img class="emoji" title=":wrench:" alt=":wrench:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f527.png" height="20" width="20" align="absmiddle"> . Make sure that the mass changes are correctly given in the <code class="highlighter-rouge">MSGFPlusAdapter</code> <img class="emoji" title=":wrench:" alt=":wrench:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f527.png" height="20" width="20" align="absmiddle"> .</li>
</ul>

<h2 id="citation">Citation</h2>

<p>If you use this workflow directly, or any derivative of it, in work leading to a scientific publication,
please cite:</p>

<p>Lai, Z.W., Weisser, J., Nilse, L., Costa, F., Keller, E., Tholen, M., Kizhakkedathu, J.N., Biniossek, M., Bronsert, P., and Schilling, O. (2016). Formalin-Fixed, Paraffin-Embedded Tissues (FFPE) as a Robust Source for the Profiling of Native and Protease-Generated Protein Amino Termini. Mol. Cell. Proteomics 15, 2203–2213.</p>


    
    <blockquote class="key_points">
        <h3> <img class="emoji" title=":grey_exclamation:" alt=":grey_exclamation:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/2755.png" height="20" width="20" align="absmiddle"> Key points</h3>

        <ul>
        
           <li>First take home message</li>
        
           <li>Second take home message</li>
        
           <li>Third take home message</li>
        
        </ul>
    </blockquote>
    

    
      <h1>Useful literature</h1>
      <p>Useful information regarding this type of analysis with descriptions and paper references for the tools used in this tutorial, and literature for this analysis techniques and interpretations can be found <a href="https://galaxyproject.github.io/training-material//topics/proteomics#references">here</a>.</p>
    

    <h3>
<img class="emoji" title=":clap:" alt=":clap:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f44f.png" height="20" width="20" align="absmiddle"> Congratulations on successfully completing this tutorial!</h3>


    <hr>
    <blockquote class="overview">
        <h3> <img class="emoji" title=":speech_balloon:" alt=":speech_balloon:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f4ac.png" height="20" width="20" align="absmiddle"> Feedback</h3>
        Please take a moment and provide your feedback on this tutorial. Your feedback will help guide and improve future revisions to this tutorial. <a href="https://tinyurl.com/GTNfeedback" target="_blank">Feedback Form</a>
    </blockquote>

    <i>This material is the result of a collaborative work. Thanks the <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a> and all the contributors
    
    !<br>
    <br>
    Found a typo? Something is wrong in this tutorial? Edit it on <a href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/tutorials/ntails/tutorial.md">GitHub</a>.</i>
</section>

        </div>
    </body>
    <script src="https://galaxyproject.github.io/training-material//assets/js/landing_page.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
</html>
