<!DOCTYPE html>
<html lang="en">
    <head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Galaxy Training!</title>
        <link rel="stylesheet" href="https://galaxyproject.github.io/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="https://galaxyproject.github.io/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="https://galaxyproject.github.io/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="https://galaxyproject.github.io/training-material/assets/css/syntax_highlighting.css">
    </head>
    <body>
        


<header>
    <nav class="navbar navbar-expand-md navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="https://galaxyproject.github.io/training-material/">
                <img src="https://galaxyproject.github.io/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="https://galaxyproject.github.io/training-material/topics/proteomics" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i> Proteomics
                        </a>
                    </li>

                    
                    <li class="nav-item">
                        <a class="nav-link" href="https://galaxyproject.github.io/training-material/topics/proteomics/slides" title="Slides for this tutorial">
                            <i class="fa fa-slideshare" aria-hidden="true"></i> Introduction slides
                        </a>
                    </li>
                    

                    
                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/docker" title="Docker image for this tutorial">
                            <i class="fa fa-ship" aria-hidden="true"></i> Docker image
                        </a>
                    </li>
                    

                    

                    

                    
                    <li class="nav-item">
                        <a class="nav-link" href="https://galaxyproject.github.io/training-material/topics/proteomics#references" title="References">
                            <i class="fa fa-book" aria-hidden="true"></i> Literature
                        </a>
                    </li>
                    

                    <li class="nav-item">
                        <a class="nav-link" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Chat on gitter">
                            <i class="fa fa-life-ring" aria-hidden="true"></i> Help
                        </a>
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/tutorials/ntails/tutorial.md">
                            <i class="fa fa-github" aria-hidden="true"></i> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <section class="tutorial">
        <h1>ntails</h1>

        <blockquote class="overview">
            <h3>Overview</h3>

            <strong><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</strong>
            <ul>
            
            <li>First question addressed during the tutorial</li>
            
            <li>Second question addressed during the tutorial</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i> Objectives</strong>
            <ul>
            
            <li>First learning objectives of the tutorial</li>
            
            <li>Second learning objectives of the tutorial</li>
            
            </ul>

            
            <strong><i class="fa fa-check-circle" aria-hidden="true"></i> Requirements</strong>
            <ul>
            
                <li>
                    
                    <a href="https://galaxyproject.github.io/training-material/topics/introduction/">Galaxy introduction</a>
                    
                </li>
            
            
            </ul>
            

            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i> Time estimation:</strong> 1h/1d</p>
        </blockquote>

        <p>N-Tails is a special Proteomics technique to analyze peptide abundancy changes of protein N-termini. Prior to the MS measurement, N-Tails enriches unmodified, as well as acetylated N-termini. Both common and “unusual” N-termini are identified, where “unusual” means that the protein N-terminus was changes.
This is best explained by an example: directly after translation, a protein has exactly one N-terminus. When a protease is cutting the protein in half, each half has its own N-terminus. While the N-terminus of the first half protein is exactly the same as the one of the full protein precursor (“native N-terminus”), the N-terminus of the second half is different (“neo-N-terminus”) and depends on the amino acid sequence where the protein was cut.
The N-Tails technique includes the use of heavy isotope dimethyl labelling.</p>

<p>The figure below illustrates the mechanism of N-Tails. It was originally published by Stefan Tholen (doctoral thesis, not available online). Further reading on N-Tails and other N-terminal techniques, see <a href="https://dx.doi.org/10.1007/978-3-7091-0885-7_5">Tholen et al., Springer Vienna, 2013</a>.</p>

<p><img src="../../images/wf_ntails_technique.png" alt="N-Tails technique"></p>

<p>The N-Tails technique was originally designed to research protease biology and has most often been used in this field. It was originally published in <a href="https://www.ncbi.nlm.nih.gov/pubmed/20208520">Kleifeld et al., Nat. Biotechnol., 2010</a>.</p>

<blockquote>
  <h3 id="-comment-interpretation-of-n-tails-results">
<i class="fa fa-commenting-o" aria-hidden="true"></i> Comment: Interpretation of N-Tails results</h3>

  <p class="comment">Be careful not to overinterprete the results of N-Tails experiments. While the technique <strong>is</strong> fit to identify direct protease substrates, it does not discriminate direct from indirect (“downstream”) effects. Thus, most of the identified N-termini will <strong>not</strong> be direct protease substrates, even if their change in protein abundance is statistically significant.
This warning is not specific for the N-Tails technique, but applies as well to other proteomic N-terminal screening techniques (e.g. COFRADIC).</p>
</blockquote>

<p>This workflow was originally built in the OpenMS framework “TOPPAS” and published in <a href="https://www.ncbi.nlm.nih.gov/pubmed/27087653">Lai, Weisser et al., MCP, 2016</a>. It was converted to OpenMS v2.1, rebuild for the Galaxy framework and tested on the original dataset by Melanie Föll. It was designed for data analysis of a three samples combined in one MS run, a technique based on dimethyl stable isotope labeling (SIL). For more information on SIL, consult <a href="https://galaxyproject.github.io/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">this tutorial</a>. The original data were generated using pre-fractionation. Thus, peptides of one biological experiment are measured in multiple consecutive MS runs (one run per fraction).</p>

<p>The figure below gives an overview of the used Galaxy nodes. For further description of the workflow, please consider the <a href="https://www.ncbi.nlm.nih.gov/pubmed/27087653">original publication</a>.</p>

<p><img src="../../images/wf_ntails_tripleDimethylLabel.png" alt="N-Tails Galaxy Workflow"></p>

<h2 id="input">Input</h2>

<p>The workflow needs three input files:</p>

<p>1) A collection of mzML files (multiple fractions of the same experiment).
  2) A FASTA protein database for the organism of interest. For more information on protein databases, consult <a href="https://galaxyproject.github.io/training-material/topics/proteomics/tutorials/database-handling/tutorial.html">this tutorial</a></p>

<h2 id="customizing-the-workflow">Customizing the Workflow</h2>

<ul>
  <li>
<em>Running the workflow on a non-prefractionated sample:</em> Simply use only one file as an input.</li>
  <li>
<em>Running the workflow on a double dimethyl labeling (only light and heavy labeling):</em> remove the third <code class="highlighter-rouge">MSGFPlusAdapter</code> <i class="fa fa-wrench" aria-hidden="true"></i> and the following <code class="highlighter-rouge">PeptideIndexer</code> <img class="emoji" title=":wrench:" alt=":wrench:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f527.png" height="20" width="20" align="absmiddle"> . Make sure that the mass changes are correctly given in the <code class="highlighter-rouge">MSGFPlusAdapter</code> <img class="emoji" title=":wrench:" alt=":wrench:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f527.png" height="20" width="20" align="absmiddle"> .</li>
</ul>

<h2 id="citation">Citation</h2>

<p>If you use this workflow directly, or any derivative of it, in work leading to a scientific publication,
please cite:</p>

<p>Lai, Z.W., Weisser, J., Nilse, L., Costa, F., Keller, E., Tholen, M., Kizhakkedathu, J.N., Biniossek, M., Bronsert, P., and Schilling, O. (2016). Formalin-Fixed, Paraffin-Embedded Tissues (FFPE) as a Robust Source for the Profiling of Native and Protease-Generated Protein Amino Termini. Mol. Cell. Proteomics 15, 2203–2213.</p>


        
        <blockquote class="key_points">
            <h3>
<i class="fa fa-key" aria-hidden="true"></i> Key points</h3>

            <ul>
                
                <li>First take home message</li>
                
                <li>Second take home message</li>
                
                <li>Third take home message</li>
                
            </ul>
        </blockquote>
        

        
        <h1>Useful literature</h1>
        <p>Useful information regarding this type of analysis with descriptions and paper references for the tools used in this tutorial, and literature for this analysis techniques and interpretations can be found <a href="https://galaxyproject.github.io/training-material/topics/proteomics#references">here</a>.</p>
        

        <h3>
<i class="fa fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

        <hr>

        <blockquote class="overview">
            <h3>
<i class="fa fa-comments-o" aria-hidden="true"></i> Feedback</h3>
            Please take a moment and provide your feedback on this tutorial. Your feedback will help guide and improve future revisions to this tutorial.
            <a href="https://tinyurl.com/GTNfeedback">Feedback Form</a>
        </blockquote>
    </section>
</div>

<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the contributors
            
            (Florian Christoph Sigloch, Björn Grüning)
            
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/tutorials/ntails/tutorial.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="https://galaxyproject.github.io/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="https://galaxyproject.github.io/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="https://galaxyproject.github.io/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="https://galaxyproject.github.io/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
</html>
