<!DOCTYPE html>
<html>
    <head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="chrome=1">
        <title>Galaxy Training!</title>
        <link media="all" href="https://netdna.bootstrapcdn.com/bootstrap/3.2.0/css/bootstrap.min.css" rel="stylesheet">
        <link rel="stylesheet" href="https://galaxyproject.github.io/training-material//assets/css/landing_page.css">
        <link rel="stylesheet" href="https://galaxyproject.github.io/training-material//assets/css/tutorial.css">
        <link rel="stylesheet" href="https://galaxyproject.github.io/training-material//assets/css/font-awesome.css">
        <link rel="stylesheet" href="https://galaxyproject.github.io/training-material//assets/css/syntax_highlighting.css">
        <meta name="viewport" content="width=device-width">
    </head>
    <body>
        <div class="wrapper">
            


<header>
    <nav class="navbar navbar-default navbar-fixed-top" role="navigation">
        <div class="container-fluid">
          <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#navbar">
              <span class="sr-only">Toggle navigation</span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
            </button>
            <a id="site-title" class="navbar-brand" href="https://wiki.galaxyproject.org/Teach/GTN">
              <img class="pull-left img-responsive" src="https://galaxyproject.github.io/training-material///assets/images/GTN.png">
            </a>
            <a id="site-title" class="navbar-brand name-holder" href="https://galaxyproject.github.io/training-material//topics/sequence-analysis/tutorials/genetic-map-rad-seq">
              RAD-Seq to construct genetic maps
            </a>
          </div>

          <div class="collapse navbar-collapse" id="navbar">

            <ul id="main-menu" class="nav navbar-nav navbar-right">
              <li><a href="https://galaxyproject.github.io/training-material/" title="Back to front page"><i class="fa fa-home"></i> Home</a></li>
              <li><a href="https://galaxyproject.github.io/training-material//topics/sequence-analysis" title="Go back to list of tutorials"><i class="fa fa-file-text-o"></i> sequence-analysis</a></li>

              
                <li><a href="https://galaxyproject.github.io/training-material//topics/sequence-analysis/slides" title="Slides for this tutorial"><i class="fa fa-graduation-cap"></i> Introduction slides</a></li>
              
              <!--<li><a href="https://galaxyproject.github.io/training-material//sequence-analysis/slides/genetic-map-rad-seq/"><i class="fa fa-file-powerpoint-o"></i> Associated slides</a></li>-->

              

              

              
                <li><a href="https://zenodo.org/record/61771#.WA9TyJN96EI" title="Links to data"><i class="fa fa-folder-open"></i> Input Dataset</a></li>
              

              
                <li><a href="https://galaxyproject.github.io/training-material//topics/sequence-analysis#references" title="References"><i class="fa fa-book"></i> Literature</a></li>
              

            </ul>
          </div>
        </div>
    </nav>
</header>

<section class="tutorial">
    <h1>RAD-Seq to construct genetic maps</h1>

    <blockquote class="overview">
        <h3>Overview</h3>

        <p class="bold"> <img class="emoji" title=":grey_question:" alt=":grey_question:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/2754.png" height="20" width="20" align="absmiddle"> Questions</p>
        <ul>
        
          <li>How to analyze RAD sequencing data for a genetic map study?</li>
        
        </ul>

        <p class="bold"> <img class="emoji" title=":dart:" alt=":dart:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f3af.png" height="20" width="20" align="absmiddle"> Objectives</p>
        <ul>
        
          <li>SNP calling from RAD sequencing data</li>
        
          <li>Find and correct haplotypes</li>
        
          <li>Create input files for genetic map building software</li>
        
        </ul>

        
        <p class="bold"> <img class="emoji" title=":heavy_check_mark:" alt=":heavy_check_mark:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/2714.png" height="20" width="20" align="absmiddle"> Requirements</p>
        <ul>
          
            <li>
                
                <a href="https://galaxyproject.github.io/training-material//topics//introduction/">Galaxy introduction</a>
                
            </li>
          
          
        </ul>
        

        <p><span class="bold"> <img class="emoji" title=":hourglass:" alt=":hourglass:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/231b.png" height="20" width="20" align="absmiddle"> Time estimation:</span> 1/2d</p>

    </blockquote>

<h1 class="no_toc" id="introduction">Introduction</h1>

<p>Original description is reachable on a <a href="http://catchenlab.life.illinois.edu/stacks/tut_gar.php">dedicated page of the official STACKS website</a>. Writers describe that they developed a genetic map in the spotted gar and present here data from a single linkage group. The gar genetic map is an F1 pseudotest cross between two parents and 94 of their F1 progeny. They took the markers that appeared in one of the linkage groups and worked backwards to provide the raw reads from all of the stacks contributing to that linkage group.</p>

<p>We here proposed to re-analyze these data at least until genotypes determination. Data are already clean so you don’t have to demultiplex it using barcode information through <code class="highlighter-rouge">Process Radtags tool</code>.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li>
<a href="#pretreatments" id="markdown-toc-pretreatments">Pretreatments</a>    <ol>
      <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
    </ol>
  </li>
  <li><a href="#building-loci-using-stacks" id="markdown-toc-building-loci-using-stacks">Building loci using STACKS</a></li>
  <li><a href="#genotypes-determination" id="markdown-toc-genotypes-determination">Genotypes determination</a></li>
</ol>

</blockquote>

<h1 id="pretreatments">Pretreatments</h1>

<h2 id="data-upload">Data upload</h2>

<p>The original data is available at <a href="https://creskolab.uoregon.edu/stacks/tutorial/stacks_samples.tar.gz">STACKS website</a> and the subset used here is findable on <a href="https://cesgo.genouest.org/resources/370/supportingdocs">CeSGO hub</a>.</p>

<blockquote>
  <h3 id="pencil2-hands-on-data-upload">
<img class="emoji" title=":pencil2:" alt=":pencil2:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/270f.png" height="20" width="20" align="absmiddle"> Hands-on: Data upload</h3>

  <ol>
    <li>Create a new history for this RAD-seq exercise. If you are not inspired, you can name it “STACKS 1.42 RAD: genetic map” for example…</li>
    <li>
      <p>Import Fasta files from parents and 20 progenies.</p>

      <blockquote>
        <h3 id="nut_and_bolt-comments">
<img class="emoji" title=":nut_and_bolt:" alt=":nut_and_bolt:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f529.png" height="20" width="20" align="absmiddle"> Comments</h3>
        <p>If you are using the <a href="https://galaxy.genouest.org">GenOuest Galaxy instance</a>, you can load the dataset using ‘Shared Data’ -&gt; ‘Data Libraries’ -&gt; ‘1 Galaxy teaching folder’ -&gt; ‘EnginesOn’ -&gt; ‘RADseq’ -&gt; ‘Genetic map’</p>
      </blockquote>

      <blockquote>
        <h3 id="bulb-tip-importing-data-via-links">
<img class="emoji" title=":bulb:" alt=":bulb:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f4a1.png" height="20" width="20" align="absmiddle"> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>Open the Galaxy Upload Manager</li>
          <li>Select <strong>Paste/Fetch Data</strong>
</li>
          <li>Paste the following links into the text field
            <ul>
              <li>https://cesgo.genouest.org/resources/375/download/female.fa</li>
              <li>https://cesgo.genouest.org/resources/376/download/male.fa</li>
              <li>https://cesgo.genouest.org/resources/377/download/progeny_1.fa</li>
              <li>https://cesgo.genouest.org/resources/378/download/progeny_2.fa</li>
              <li>https://cesgo.genouest.org/resources/379/download/progeny_3.fa</li>
              <li>https://cesgo.genouest.org/resources/380/download/progeny_4.fa</li>
              <li>https://cesgo.genouest.org/resources/381/download/progeny_5.fa</li>
              <li>https://cesgo.genouest.org/resources/382/download/progeny_6.fa</li>
              <li>https://cesgo.genouest.org/resources/383/download/progeny_7.fa</li>
              <li>https://cesgo.genouest.org/resources/384/download/progeny_8.fa</li>
              <li>https://cesgo.genouest.org/resources/385/download/progeny_9.fa</li>
              <li>https://cesgo.genouest.org/resources/386/download/progeny_10.fa</li>
              <li>https://cesgo.genouest.org/resources/387/download/progeny_11.fa</li>
              <li>https://cesgo.genouest.org/resources/388/download/progeny_12.fa</li>
              <li>https://cesgo.genouest.org/resources/389/download/progeny_13.fa</li>
              <li>https://cesgo.genouest.org/resources/390/download/progeny_14.fa</li>
              <li>https://cesgo.genouest.org/resources/391/download/progeny_15.fa</li>
              <li>https://cesgo.genouest.org/resources/392/download/progeny_16.fa</li>
              <li>https://cesgo.genouest.org/resources/393/download/progeny_17.fa</li>
              <li>https://cesgo.genouest.org/resources/394/download/progeny_18.fa</li>
              <li>https://cesgo.genouest.org/resources/395/download/progeny_19.fa</li>
              <li>https://cesgo.genouest.org/resources/417/download/progeny_20.fa</li>
            </ul>
          </li>
          <li>Press <strong>Start</strong>
</li>
        </ul>
      </blockquote>

      <p>As default, Galaxy takes the link as name. It also do not link the dataset to a database or a reference genome.</p>
    </li>
  </ol>

</blockquote>

<h1 id="building-loci-using-stacks">Building loci using STACKS</h1>

<p>Run <code class="highlighter-rouge">Stacks: De novo map</code> Galaxy tool. This program will run <code class="highlighter-rouge">ustacks</code>, <code class="highlighter-rouge">cstacks</code>, and <code class="highlighter-rouge">sstacks</code> on each individual, accounting for the alignments of each read.</p>

<blockquote>
  <h3 id="nut_and_bolt-comment">
<img class="emoji" title=":nut_and_bolt:" alt=":nut_and_bolt:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f529.png" height="20" width="20" align="absmiddle"> Comment</h3>

  <p>Information on <code class="highlighter-rouge">denovo_map.pl</code> and its parameters can be found online: https://creskolab.uoregon.edu/stacks/comp/denovo_map.php.</p>
</blockquote>

<blockquote>
  <p><strong>Stacks: De novo map</strong> <img class="emoji" title=":wrench:" alt=":wrench:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f527.png" height="20" width="20" align="absmiddle">: Run Stacks selecting the Genetic map usage. Specify each parent as a sample in the appropriate box, then each of the 20 progenies and specify a CP Cross type, 3 for the Minimum number of identical raw reads required to create a stack, 3 for minimum number of identical raw reads required to create a stack in ‘progeny’ individuals, 3 for the number of mismatches allowed between loci when building the catalog and activate the option “remove, or break up, highly repetitive RAD-Tags in the ustacks program”.</p>

  <p><img src="../../images/RAD2_Genetic_Map/denovomap_in.png" alt=""></p>
</blockquote>

<blockquote>
  <p>Once Stacks has completed running, you will see 5 new data collections and 8 datasets.</p>

  <p><img src="../../images/RAD2_Genetic_Map/denovomap_out.png" alt=""></p>

  <p>Investigate the output files: <code class="highlighter-rouge">result.log</code> and <code class="highlighter-rouge">catalog.*</code> (snps, alleles and tags).</p>

  <p>Looking at the first file, <code class="highlighter-rouge">denovo_map.log</code>, you can see the command line used and the start as end execution time.</p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_log_top.png" alt=""></p>

  <p>Then are the different STACKS steps:</p>

  <p><code class="highlighter-rouge">ustacks</code></p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_log_ustacks.png" alt=""></p>

  <p><code class="highlighter-rouge">cstacks</code></p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_log_cstacks.png" alt=""></p>

  <blockquote>
    <h3 id="question-question">
<img class="emoji" title=":question:" alt=":question:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/2753.png" height="20" width="20" align="absmiddle"> Question</h3>

    <ol>
      <li>Can you identify the meanning of the number 425?</li>
      <li>
        <p>Looking at the catalog.tags file, identify specific and shared loci from each individual. Count nuber of catalog loci coming from the first individual, from the second and thus find on both parents.</p>

        <details>
<summary>Click to view answers</summary>
<ol type="1">
<li> Here, the catalog is made with 459 tags, 425 coming from the "reference individual", here female. Some of these 425 can be shared with the other parent.</li>
<li>3500</li>
</ol>
</details>
      </li>
    </ol>
  </blockquote>

  <p><code class="highlighter-rouge">sstacks</code></p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_log_sstacks.png" alt=""></p>

  <p>Endly, <code class="highlighter-rouge">genotypes</code> is executed. It search for markers identified on the parents and the associate progenies haplotypes. If the first parent have a GA (ex: aatggtgtGgtccctcgtAc) and AC (ex: aatggtgtAgtccctcgtCc) haplotypes, and the second parent only a GA (ex: aatggtgtGgtccctcgtAc) haplotype, STACKS declare a ab/aa marker for this locus. Genotypes program then associate GA to a and AC to b and then scan progeny to determine which haplotype is found on each of them.</p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_log_genotypes1.png" alt=""></p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_log_genotypes2.png" alt=""></p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_log_end.png" alt=""></p>

  <p>Finally, 447 loci, markers, are kept to generate the <code class="highlighter-rouge">batch_1.genotypess_1.tsv</code> file. 459 loci are stored on the observed haplotype file <code class="highlighter-rouge">batch_1.haplotypes_1.tsv</code>.</p>

</blockquote>

<h3 id="matches-files">Matches files</h3>
<blockquote>

  <p>Here are <code class="highlighter-rouge">sample1.snps</code> (left) and <code class="highlighter-rouge">sample2.snps</code> (right)</p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_matches1.PNG" alt=""></p>

  <p>Catalog_ID (= catalog Stacks_ID) is composed by the <code class="highlighter-rouge">Stack_ID</code> from the “reference” individual, here sample1, but number is different from sample2 <code class="highlighter-rouge">Stack_ID</code>. Thus, in the <code class="highlighter-rouge">catalog.alleles.tsv</code>, the <code class="highlighter-rouge">Stack_ID</code> numbered 3 correspond to the <code class="highlighter-rouge">Stack_ID</code> number 16 from sample2!</p>

  <p>Here you can observe matches files (you maybe have to change the tsv datatype into a tabular one to display correctly the datasets).</p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_matches2.png" alt=""></p>

  <p>Conidering catalog SNPs 27 &amp; 28, on the 302 catalog locus:</p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_matches_snps.png" alt=""></p>

  <p>and corresponding catalog haplotypes, 3 on the 4 possible (AA, AT, GT but no GA):</p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_matches_alleles_haplotypes.png" alt=""></p>

  <p>heterozygoty is observed on each parent (one ab, the other ac) and we have 19 genotypes on the 22 individuals.</p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_matches_markers.png" alt=""></p>

  <p>We then can see that Stack_ID 330 for female correspond to the 39 for male:</p>

  <p><img src="../../images/RAD2_Genetic_Map/denovo_map_matches_alleles_male_female.png" alt=""></p>

</blockquote>

<h1 id="genotypes-determination">Genotypes determination</h1>
<blockquote>
  <p><strong>Stacks: genotypes</strong> <img class="emoji" title=":wrench:" alt=":wrench:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f527.png" height="20" width="20" align="absmiddle">: Re-Run the last step of <code class="highlighter-rouge">Stacks: De novo map</code> pipeline specifying more options as:</p>
  <blockquote>
    <ol>
      <li>
        <p>The genetic map type (ie F1, F2 (left figure, F1xF1), Double Haploid, Back Cross (F1xF0), Cross Pollination (right figure, F1 or F2 but resulting from the croos of pure homozygous parents))</p>

        <p><img src="../../images/RAD2_Genetic_Map/Genetic_map_F2.png" alt="">    <img src="../../images/RAD2_Genetic_Map/Genetic_map_CrossPollination.png" alt=""></p>
      </li>
      <li>
        <p>Genotyping informations export format for input in genetic mapper tools (ie JoinMap, R/qtl, …). To be notice that R/qtl format for an F2 cross type can be an input for MapMaker or Carthagene.</p>
      </li>
      <li>
        <p>Tresholds concernaing a minimal number of progeny and/or minimum stacks depth to consider a locus</p>
      </li>
      <li>
        <p>Automatic corrections. Concerning this last option, it’s possible to to ask genotypes making automatic corrections on some errors as homozygous tags verification in the progeny to be sure a SNP is missing. Thus, if SNP detection model can’t identify a site as heterygous or homozygous, this site is temporarily tagged as homozygous to facilitate the search, by sstacks, of concordance with the loci catalog. If a second allele is detected on the catalog (ie, in parents) is found on a progeny with a weak frequency (&lt;10% of a stack reads number), genotypes program can correct the genotype. Additionnaly, it will delete a homozygous genotype on a particular individual if locus genotype is supported by less than 5 reads. Thus corrected genotypes are marked uppercase.</p>
      </li>
    </ol>
  </blockquote>

  <p>Here is an example of a locus originally marke as homozygous before automatic correction because on allele is supported by less than 5 reads. After correction, this locus is marked as heterygous.</p>

  <p><img src="../../images/RAD2_Genetic_Map/genotypes_automatic_correction.png" alt=""></p>

  <p>You can re-run <strong>Stacks: genotypes</strong> <img class="emoji" title=":wrench:" alt=":wrench:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f527.png" height="20" width="20" align="absmiddle">: program modifying the number of genotyped progeny to consider a marker and thus be more or less stringent. Compare results.</p>

  <h3 id="genotypestsv-files">Genotypes.tsv files</h3>

  <p>One line by locus, one column by individual (aa, ab, AB if automatic correction applied, bb, bc, …) with observed genotype for each locus:</p>

  <p><img src="../../images/RAD2_Genetic_Map/genotypes_tsv.png" alt=""></p>

  <h3 id="genotypestxt-files">Genotypes.txt files</h3>

  <p>One line by individual, and for each individual, for each catalog locus, genotype:</p>

  <p><img src="../../images/RAD2_Genetic_Map/genotypes_txt.png" alt=""></p>

  <h3 id="haplotypestsv-files">Haplotypes.tsv files</h3>

  <p>One line by locus, one column by individual (aa, ab, AB if automatic correction applied, bb, bc, …) with observed genotype for each locus:</p>

  <p><img src="../../images/RAD2_Genetic_Map/haplotypes_tsv.png" alt=""></p>
</blockquote>

<blockquote>

  <blockquote>
    <h3 id="question-question-1">
<img class="emoji" title=":question:" alt=":question:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/2753.png" height="20" width="20" align="absmiddle"> Question</h3>

    <ol>
      <li>The use of the deleverage algorithm allows to not consider loci obtained from merging more than 3 stacks. Why 3 if biologically, you are waiting something related to 2 for diploid organisms?</li>
      <li>
        <p>Re-execute <strong>Stacks: De novo map</strong> pipeline modifying the p-value treshold for the SNP model. What is the difference regarding to unverified haplotypes ?</p>

        <details>
<summary>Click to view answers</summary>
<ol type="1">
<li>This value of 3 is important to use if we don't want to blacklist loci for whom 99.9% of individuals have one and/or the alt allele and 0.01% have a third one, resulting of a sequencing error.</li>
<li>We see a moficiation of the number of unverified haplotypes</li>
</ol>
</details>
      </li>
    </ol>
  </blockquote>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>In this tutorial, we have analyzed real RAD sequencing data to extract useful information, such as genotypes and haplotypes to generate input files for downstream genetic map creation. This approach can be sum up with the following scheme:</p>

<p><img src="../../images/genetic_map_workflow.PNG" alt=""></p>


    
    <blockquote class="key_points">
        <h3> <img class="emoji" title=":grey_exclamation:" alt=":grey_exclamation:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/2755.png" height="20" width="20" align="absmiddle"> Key points</h3>

        <ul>
        
           <li>Simple sentence to sum up the first key point of the tutorial (Take home message)</li>
        
           <li>Second key point</li>
        
           <li>Third key point</li>
        
           <li>...</li>
        
        </ul>
    </blockquote>
    

    

    <h3>
<img class="emoji" title=":clap:" alt=":clap:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f44f.png" height="20" width="20" align="absmiddle"> Congratulations on successfully completing this tutorial!</h3>


    <hr>
    <blockquote class="overview">
        <h3> <img class="emoji" title=":speech_balloon:" alt=":speech_balloon:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f4ac.png" height="20" width="20" align="absmiddle"> Feedback</h3>
        Please take a moment and provide your feedback on this tutorial. Your feedback will help guide and improve future revisions to this tutorial. <a href="https://tinyurl.com/GTNfeedback" target="_blank">Feedback Form</a>
    </blockquote>

    <i>This material is the result of a collaborative work. Thanks to <a href="https://galaxyproject.github.io/training-material//sequence-analysis#contributors">all the contributors</a> and the <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>!<br>
    Found a typo? Something is wrong in this tutorial? Edit it on <a href="https://github.com/galaxyproject/training-material/tree/master/sequence-analysis/tutorials/genetic-map-rad-seq.md">GitHub</a>.</i>
</section>

        </div>
    </body>
    <script src="https://galaxyproject.github.io/training-material//assets/js/landing_page.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
</html>
