<!DOCTYPE html>
<html>
    <head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="chrome=1">
        <title>Galaxy Training!</title>
        <link media="all" href="https://netdna.bootstrapcdn.com/bootstrap/3.2.0/css/bootstrap.min.css" rel="stylesheet">
        <link rel="stylesheet" href="https://galaxyproject.github.io/training-material//assets/css/landing_page.css">
        <link rel="stylesheet" href="https://galaxyproject.github.io/training-material//assets/css/tutorial.css">
        <link rel="stylesheet" href="https://galaxyproject.github.io/training-material//assets/css/font-awesome.css">
        <link rel="stylesheet" href="https://galaxyproject.github.io/training-material//assets/css/syntax_highlighting.css">
        <meta name="viewport" content="width=device-width">
    </head>
    <body>
        <div class="wrapper">
            


<header>
    <nav class="navbar navbar-default navbar-fixed-top" role="navigation">
        <div class="container-fluid">
          <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#navbar">
              <span class="sr-only">Toggle navigation</span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
            </button>
            <a id="site-title" class="navbar-brand" href="https://wiki.galaxyproject.org/Teach/GTN">
              <img class="pull-left img-responsive" src="https://galaxyproject.github.io/training-material///assets/images/GTN.png">
            </a>
            <a id="site-title" class="navbar-brand name-holder" href="https://galaxyproject.github.io/training-material//topics/usegalaxy/tutorials/collections/tutorial.html">
              Using dataset collection
            </a>
          </div>

          <div class="collapse navbar-collapse" id="navbar">

            <ul id="main-menu" class="nav navbar-nav navbar-right">
              <li><a href="https://galaxyproject.github.io/training-material/" title="Back to front page"><i class="fa fa-home"></i> Home</a></li>
              <li><a href="https://galaxyproject.github.io/training-material//topics/usegalaxy" title="Go back to list of tutorials"><i class="fa fa-file-text-o"></i> Tutorials using public Galaxy at usegalaxy.org</a></li>

              
                <li><a href="https://galaxyproject.github.io/training-material//topics/usegalaxy/slides" title="Slides for this tutorial"><i class="fa fa-graduation-cap"></i> Introduction slides</a></li>
              
              <!--<li><a href="https://galaxyproject.github.io/training-material//usegalaxy/slides/collections/"><i class="fa fa-file-powerpoint-o"></i> Associated slides</a></li>-->

              
                <li><a href="https://github.com/galaxyproject/training-material/tree/master/topics/usegalaxy/docker" title="Docker image for this tutorial"><i class="fa fa-ship"></i> Docker image</a></li>
              

              
                <li><a href="https://usegalaxy.org" title="Galaxy instance for this tutorial"><i class="fa fa-external-link"></i> Galaxy instance</a></li>
              

              

              

            </ul>
          </div>
        </div>
    </nav>
</header>

<section class="tutorial">
    <h1>Using dataset collection</h1>

    <blockquote class="overview">
        <h3>Overview</h3>

        <p class="bold"> <img class="emoji" title=":grey_question:" alt=":grey_question:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/2754.png" height="20" width="20" align="absmiddle"> Questions</p>
        <ul>
        
          <li>How to manipulate large numbers of datasets at once?</li>
        
        </ul>

        <p class="bold"> <img class="emoji" title=":dart:" alt=":dart:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f3af.png" height="20" width="20" align="absmiddle"> Objectives</p>
        <ul>
        
          <li>Understand and master dataset collections</li>
        
        </ul>

        
        <p class="bold"> <img class="emoji" title=":heavy_check_mark:" alt=":heavy_check_mark:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/2714.png" height="20" width="20" align="absmiddle"> Requirements</p>
        <ul>
          
            <li>
                
                <a href="https://galaxyproject.github.io/training-material//topics//introduction/">Galaxy introduction</a>
                
            </li>
          
          
        </ul>
        

        <p><span class="bold"> <img class="emoji" title=":hourglass:" alt=":hourglass:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/231b.png" height="20" width="20" align="absmiddle"> Time estimation:</span> 30 min</p>

    </blockquote>

<p>Here we will show Galaxy features designed to help with the analysis of large numbers of samples. When you have just a few samples - clicking through them is easy. But once you’ve got hundreds - it becomes very annoying. In Galaxy we have introduced <strong>Dataset collections</strong> that allow you to combine numerous datasets in a single entity that can be easily manipulated.</p>

<h1 id="getting-data">Getting data</h1>
<p><a href="https://usegalaxy.org/library/list#folders/Fab5f788f07073c11">Here</a> is a history containing a few datasets we will be practicing with (as always with Galaxy tutorial you can upload your own data and play with it instead of the provided datasets):</p>

<ul>
  <li>
<code class="highlighter-rouge">M117-bl_1</code> - family 117, mother, 1-st (<strong>F</strong>) read from <strong>blood</strong>
</li>
  <li>
<code class="highlighter-rouge">M117-bl_2</code> - family 117, mother, 2-nd (<strong>R</strong>) read from <strong>blood</strong>
</li>
  <li>
<code class="highlighter-rouge">M117-ch_1</code> - family 117, mother, 1-st (<strong>F</strong>) read from <strong>cheek</strong>
</li>
  <li>
<code class="highlighter-rouge">M117-ch_1</code> - family 117, mother, 2-nd (<strong>R</strong>) read from <strong>cheek</strong>
</li>
  <li>
<code class="highlighter-rouge">M117C1-bl_1</code>- family 117, child, 1-st (<strong>F</strong>) read from <strong>blood</strong>
</li>
  <li>
<code class="highlighter-rouge">M117C1-bl_2</code>- family 117, child, 2-nd (<strong>R</strong>) read from <strong>blood</strong>
</li>
  <li>
<code class="highlighter-rouge">M117C1-ch_1</code>- family 117, child, 1-st (<strong>F</strong>) read from <strong>cheek</strong>
</li>
  <li>
<code class="highlighter-rouge">M117C1-ch_2</code>- family 117, child, 2-nd (<strong>R</strong>) read from <strong>cheek</strong>
</li>
</ul>

<p>These datasets represent genomic DNA (enriched for mitochondria via a long range PCR) isolated from blood (<code class="highlighter-rouge">bl</code>) and cheek (buccal swab, <code class="highlighter-rouge">ch</code>) of mother (<code class="highlighter-rouge">M117</code>) and her child (<code class="highlighter-rouge">M117C1</code>) that was sequenced on an Illumina miSeq machine as paired-read library (250-bp reads; see our <a href="https://www.pnas.org/content/111/43/15474.abstract">2014</a> manuscript for <strong>Methods</strong>).</p>

<h2 id="load-data-from-galaxy-library">Load data from Galaxy library</h2>

<p>Right click (or Ctrl-click) on <a href="https://usegalaxy.org/library/list#folders/Fab5f788f07073c11">this link</a> to open a new browser window and position this window side-by-side with the window displaying this tutorial. You will see this:</p>

<p class="img-responsive"><img src="../../images/grab_data.png" alt=""></p>

<p>Select all datasets and click <strong>to History</strong> button. This will import all datasets into a history. Follow the direction and will see a screen like this:</p>

<figure id="figure-1" class="img-responsive"><img src="../../images/manyDatasets.png" alt="manyDatasets" title="manyDatasets"><figcaption><span class="figcaption-prefix">Figure 1:</span> manyDatasets</figcaption></figure>

<h2 id="creating-a-paired-dataset-collection">Creating a paired dataset collection</h2>

<p>Now click the checkbox in <img src="../../images/historyItemControls.png" alt=""> and you will see your history changing like this:</p>

<figure id="figure-2" class="img-responsive"><img src="../../images/historyWithCheckboxes.png" alt="historyWithCheckboxes" title="historyWithCheckboxes"><figcaption><span class="figcaption-prefix">Figure 2:</span> historyWithCheckboxes</figcaption></figure>

<p>Let’s click <strong>All</strong>, which will select all datasets in the history, then click <strong>For all selected…</strong> and finally select <strong>Build List of Dataset Pairs</strong> from the following menu:</p>

<figure id="figure-3" class="img-responsive"><img src="../../images/buildPairs.png" alt="buildPairs" title="buildPairs"><figcaption><span class="figcaption-prefix">Figure 3:</span> buildPairs</figcaption></figure>

<p>The following wizard will appear:</p>

<figure id="figure-4" class="img-responsive"><img src="../../images/collectionCreation.png" alt="collectionCreation" title="collectionCreation"><figcaption><span class="figcaption-prefix">Figure 4:</span> collectionCreation</figcaption></figure>

<p>In this case Galaxy automatically assigned pairs using the <code class="highlighter-rouge">_1</code> and <code class="highlighter-rouge">_2</code> endings of dataset names. Let’s however pretend that this did not happen. Click on <strong>Unpair all</strong> (highlighted in red in the figure above) link and then on <strong>Clear</strong> link (highlighted in blue in the figure above). The interface will change into its virgin state:</p>

<figure id="figure-5" class="img-responsive"><img src="../../images/collectionCreationClean.png" alt="collectionCreationClean" title="collectionCreationClean"><figcaption><span class="figcaption-prefix">Figure 5:</span> collectionCreationClean</figcaption></figure>

<p>Hopefully you remember that we have paired-end data in this scenario. Datasets containing the first (forward) and the second (reverse) read are differentiated by having <code class="highlighter-rouge">_1</code> and <code class="highlighter-rouge">_2</code> in the filename. We can use this feature in dataset collection wizard to pair our datasets.  Type <code class="highlighter-rouge">_1</code> in the left <strong>Filter this list</strong> text box and <code class="highlighter-rouge">_2</code> in the right:</p>

<figure id="figure-6" class="img-responsive"><img src="../../images/1and2.png" alt="1and2" title="1and2"><figcaption><span class="figcaption-prefix">Figure 6:</span> 1and2</figcaption></figure>

<p>You will see that the dataset collection wizard will automatically filter lists on each side of the interface:</p>

<figure id="figure-7" class="img-responsive"><img src="../../images/collectionPrefiltered.png" alt="collectionPrefiltered" title="collectionPrefiltered"><figcaption><span class="figcaption-prefix">Figure 7:</span> collectionPrefiltered</figcaption></figure>

<p>Now you can either click <strong>Auto pair</strong> if pairs look good to you (proper combinations of datasets are listed in each line) or pair each forward/reverse group individually by pressing <strong>Pair these datasets</strong> button separating each pair:</p>

<figure id="figure-8" class="img-responsive"><img src="../../images/collectionCreation.png" alt="collectionCreation" title="collectionCreation"><figcaption><span class="figcaption-prefix">Figure 8:</span> collectionCreation</figcaption></figure>

<p>Now it is time to name the collection:</p>

<figure id="figure-9"><img src="../../images/collectionNaming.png" alt="collectionNaming" title="collectionNaming"><figcaption><span class="figcaption-prefix">Figure 9:</span> collectionNaming</figcaption></figure>

<p>and create the collection by clicking <strong>Create list</strong>. A new item will appear in the history as you can see on the panel <strong>A</strong> below. Clicking on collection will expand it to show four pairs it contains (panel <strong>B</strong>). Clicking individual pairs will expand them further to reveal <strong>forward</strong> and <strong>reverse</strong> datasets (panel <strong>C</strong>). Expanding these further will enable one to see individual datasets (panel <strong>D</strong>).</p>

<figure id="figure-10" class="img-responsive"><img src="../../images/collection_ABCD.png" alt="collection_ABCD" title="collection_ABCD"><figcaption><span class="figcaption-prefix">Figure 10:</span> collection_ABCD</figcaption></figure>

<h2 id="using-collections">Using collections</h2>

<p>By now we see that a collection can be used to bundle a large number of items into a single history item. This means that many Galaxy tools will be able to process all datasets in a collection transparently to you. Let’s try to map these datasets to human genome using <code class="highlighter-rouge">bwa-mem</code> mapper:</p>

<figure id="figure-11" class="img-responsive"><img src="../../images/bwa_mem_collection_readGroups.png" alt="bwa_mem_collection_readGroups" title="bwa_mem_collection_readGroups"><figcaption><span class="figcaption-prefix">Figure 11:</span> bwa_mem_collection_readGroups</figcaption></figure>

<p>Here is what you need to do:</p>

<ul>
  <li>set <strong>Using reference genome</strong> to <code class="highlighter-rouge">hg38</code> (red outline);</li>
  <li>set <strong>Single or Paired-end reads</strong> to <code class="highlighter-rouge">Paired collection</code> (blue outline);</li>
  <li>select <code class="highlighter-rouge">M177-collection</code> from <strong>Select a paired collection</strong> dropdown (magenta outline);</li>
  <li>In <strong>Set read groups information</strong> select <code class="highlighter-rouge">Automatically assign ID</code> (green outline);</li>
  <li>scroll down and click <strong>Execute</strong>.</li>
</ul>

<p>You will see jobs being submitted and new datasets appearing in the history. IN particular below you can see that Galaxy has started four jobs (two yellow and two gray). This is because we have eight paired datasets with each pair being processed separately by <code class="highlighter-rouge">bwa-mem</code>. As a result we have four <code class="highlighter-rouge">bwa-mem</code> runs:</p>

<figure id="figure-12" class="img-responsive"><img src="../../images/bwa_memCollectionRunning.png" alt="bwa_memCollectionRunning" title="bwa_memCollectionRunning"><figcaption><span class="figcaption-prefix">Figure 12:</span> bwa_memCollectionRunning</figcaption></figure>

<p>Once these jobs are finished they will disappear from the history and all results will be represented as a new collection:</p>

<figure id="figure-13" class="img-responsive"><img src="../../images/bwa_memCollectionDone.png" alt="bwa_memCollectionDone" title="bwa_memCollectionDone"><figcaption><span class="figcaption-prefix">Figure 13:</span> bwa_memCollectionDone</figcaption></figure>

<p>Let’s look at this collection by clicking on it (panel <strong>A</strong> in the figure below). You can see that now this collection is no longer paired (compared to the collection we created in the beginning of this tutorial). This is because <code class="highlighter-rouge">bwa-mem</code> takes forward and reverse data as input, but produces only a single BAM dataset as the output. So what we have in the result is a <em>list</em> of four dataset (BAM files; panels <strong>B</strong> and <strong>C</strong>).</p>

<figure id="figure-14" class="img-responsive"><img src="../../images/bwa_memCollection_ABC.png" alt="bwa_memCollection_ABC" title="bwa_memCollection_ABC"><figcaption><span class="figcaption-prefix">Figure 14:</span> bwa_memCollection_ABC</figcaption></figure>

<h2 id="processing-collection-as-a-single-entity">Processing collection as a single entity</h2>

<p>Now that <code class="highlighter-rouge">bwa-mem</code> has finished and generated a collection of BAM datasets we can continue to analyze the entire collection as a single Galaxy ‘<em>item</em>’.</p>

<h3 id="ensuring-consistency-of-bam-dataset">Ensuring consistency of BAM dataset</h3>

<p>Let’s perform cleanup of our BAM files with <code class="highlighter-rouge">cleanSam</code> utility from the <strong>Picard</strong> package:</p>

<figure id="figure-15" class="img-responsive"><img src="../../images/cleanSam.png" alt="cleanSam" title="cleanSam"><figcaption><span class="figcaption-prefix">Figure 15:</span> cleanSam</figcaption></figure>

<p>If you look at the picture above carefully, you will see that the <strong>Select SAM/BAM dataset or dataset collection</strong> parameter is empty (it says <code class="highlighter-rouge">No sam or bam datasets available.</code>). This is because we do not have single SAM or BAM datasets in the history. Instead we have a collection. So all you need to do is to click on the <strong>folder</strong> <img src="../images/folder.png"> button and you will get our BAM collection selected:</p>

<figure id="figure-16" class="img-responsive"><img src="../../images/cleanSam_closeup.png" alt="cleanSam_closeup" title="cleanSam_closeup"><figcaption><span class="figcaption-prefix">Figure 16:</span> cleanSam_closeup</figcaption></figure>

<p>Click <strong>Execute</strong>. As an output this tool will produce a collection contained cleaned data.</p>

<h3 id="retaining-proper-pairs">Retaining ‘proper pairs’</h3>

<p>Now let’s clean the dataset further by only preserving truly paired reads (reads satisfying two requirements: (1) read is paired, and (2) it is mapped as a proper pair). For this we will use <code class="highlighter-rouge">Filter SAM or BAM</code> tools from <strong>SAMTools</strong> collection:</p>

<figure id="figure-17" class="img-responsive"><img src="../../images/filter.png" alt="filter" title="filter"><figcaption><span class="figcaption-prefix">Figure 17:</span> filter</figcaption></figure>

<p>parameters should be set as shown below. By setting mapping quality to <code class="highlighter-rouge">20</code> we avoid reads mapping to multiple locations and by using <strong>Filter on bitwise flag</strong> option we ensure that the resulting dataset will contain only properly paired reads. This operation will produce yet another collection containing now filtered datasets.</p>

<figure id="figure-18" class="img-responsive"><img src="../../images/filter_closeup.png" alt="filter_closeup" title="filter_closeup"><figcaption><span class="figcaption-prefix">Figure 18:</span> filter_closeup</figcaption></figure>

<h3 id="merging-collection-into-a-single-dataset">Merging collection into a single dataset</h3>

<p>The beauty of BAM datasets is that they can be combined in a single entity using so called <em>Read group</em>. This allows to bundle reads from multiple experiments into a single dataset where read identity is maintained by labelling every sequence with <em>read group</em> tags. So let’s finally reduce this collection to a single BAM dataset. For this we will use <code class="highlighter-rouge">MergeSamFiles</code> tool for the <code class="highlighter-rouge">Picard</code> suite:</p>

<figure id="figure-19" class="img-responsive"><img src="../../images/merge.png" alt="merge" title="merge"><figcaption><span class="figcaption-prefix">Figure 19:</span> merge</figcaption></figure>

<p>Here we select the collection generated by the filtering tool described above:</p>

<figure id="figure-20" class="img-responsive"><img src="../../images/merge_closeup.png" alt="merge_closeup" title="merge_closeup"><figcaption><span class="figcaption-prefix">Figure 20:</span> merge_closeup</figcaption></figure>

<p>This operation will <strong>not</strong> generate a collection. Instead, it will generate a single BAM dataset containing mapped reads from our four samples (<code class="highlighter-rouge">M117-bl</code>, <code class="highlighter-rouge">M117-ch</code>, <code class="highlighter-rouge">M117C1-bl</code>, and <code class="highlighter-rouge">M117C1-ch</code>).</p>

<h2 id="lets-look-at-what-weve-got">Let’s look at what we’ve got!</h2>

<p>So we have one BAM dataset combining everything we’ve done so far. Let’s look at the contents of this dataset using a genome browser. First, we will need to downsample the dataset to avoiding overwhelming the browser. For this we will use <code class="highlighter-rouge">Downsample SAM/BAM</code> tool:</p>

<figure id="figure-21" class="img-responsive"><img src="../../images/downsample.png" alt="downsample" title="downsample"><figcaption><span class="figcaption-prefix">Figure 21:</span> downsample</figcaption></figure>

<p>Set <strong>Probability (between 0 and 1) that any given read will be kept</strong> to roughly <code class="highlighter-rouge">5%</code> (or <code class="highlighter-rouge">0.05</code>) using the slider control:</p>

<figure id="figure-22"><img src="../../images/downsample_closeup.png" alt="downsample_closeup" title="downsample_closeup"><figcaption><span class="figcaption-prefix">Figure 22:</span> downsample_closeup</figcaption></figure>

<p>This will generate another BAM dataset containing only 5% of the original reads and much smaller as a result. Click on this dataset and you will see links to various genome browsers:</p>

<figure id="figure-23" class="img-responsive"><img src="../../images/browserLinks.png" alt="browserLinks" title="browserLinks"><figcaption><span class="figcaption-prefix">Figure 23:</span> browserLinks</figcaption></figure>

<p>Click the <strong>Human hg38</strong> link in the <strong>display with IGV</strong> line as highlighted above (to learn more about displaying Galaxy data in IGV with this <a href="https://vimeo.com/123442619#t=4m16s">movie</a>). Below is an example generated with IGV on these data. In this screenshot reads are colored by read group (four distinct colors). A yellow inset displays additional information about a single read. One can see that this read corresponds to read group <code class="highlighter-rouge">M117-bl</code>.</p>

<figure id="figure-24" class="img-responsive"><img src="../../images/igv.png" alt="igv" title="igv"><figcaption><span class="figcaption-prefix">Figure 24:</span> igv</figcaption></figure>

<h2 id="we-did-not-fake-this">We did not fake this:</h2>
<p>The history described in this page is accessible directly from here:</p>

<ul>
  <li>History <a href="https://usegalaxy.org/u/aun1/h/collections-full-analysis"><strong>Collections (full analysis)</strong></a>
</li>
</ul>

<p>From there you can import histories to make them your own.</p>

<h2 id="if-things-dont-work">If things don’t work…</h2>
<p>…you need to complain. Use <a href="https://usegalaxy.org/biostar/biostar_redirect">Galaxy’s BioStar Channel</a> to do this.</p>


    

    

    <h3>
<img class="emoji" title=":clap:" alt=":clap:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f44f.png" height="20" width="20" align="absmiddle"> Congratulations on successfully completing this tutorial!</h3>


    <hr>
    <blockquote class="overview">
        <h3> <img class="emoji" title=":speech_balloon:" alt=":speech_balloon:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f4ac.png" height="20" width="20" align="absmiddle"> Feedback</h3>
        Please take a moment and provide your feedback on this tutorial. Your feedback will help guide and improve future revisions to this tutorial. <a href="https://tinyurl.com/GTNfeedback" target="_blank">Feedback Form</a>
    </blockquote>

    <i>This material is the result of a collaborative work. Thanks the <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a> and all the contributors
    
    !<br>
    <br>
    Found a typo? Something is wrong in this tutorial? Edit it on <a href="https://github.com/galaxyproject/training-material/tree/master/topics/usegalaxy/tutorials/collections/tutorial.md">GitHub</a>.</i>
</section>

        </div>
    </body>
    <script src="https://galaxyproject.github.io/training-material//assets/js/landing_page.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
</html>
