---
layout: tutorial_slides
topic_name: "Dev-Corner"
tutorial_name: tool_integration
logo: "GTN"
---

# Tool development and integration into Galaxy

---




<!-- _____________ INTRODUCTION _____________ -->

## A tool in Galaxy

---

### A tool in Galaxy

![](../images/galaxy_instance_detailed_screenshot.png)

---

### Galaxy tool / wrapper

![](../images/graphlan_screenshot.png)

```bash
graphlan.py --format "png" --size 7 input_tree.txt png_image.png
```

---

### Wrapper

- Description of the user interface
- Link between the tool and Galaxy instance
- How to invoke the tool
- What options to pass
- What files the tool will produce as output

---

### Wrapper

![](../images/wrapper_layers.png)

---

![](../images/planemo-logo.png)

> Command-line utilities to assist in building and publishing Galaxy tools.

- [Documentation](https://planemo.readthedocs.io/en/latest/)
- [Tutorial](https://planemo.readthedocs.io/en/latest/writing_standalone.html)

---

### <i class="fa fa-question-circle" aria-hidden="true"></i> Questions

- What is a tool for Galaxy?
- How to build a tool/wrapper with the good practices?
- How to deal with the tool environment?

---

### <i class="fa fa-bullseye" aria-hidden="true"></i> Objectives

- Discover what is a wrapper and its structure
- Use the Planemo utilities to develop a good wrapper
- Deal with the dependencies
- Write functional tests
- Make a tool ready for publishing in a ToolShed

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

![](../images/exercise.png)

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### The Goal

- Developing a wrapper for Seqtk

    A tool for processing sequence data in FASTA and FASTQ files

---




<!-- _____________ WRAPPER _____________ -->

## Wrapper

---

### Wrapper

![](../images/wrapper_big_picture_1.png)

---

### Wrapper

![](../images/wrapper_big_picture_2.png)

---

### Wrapper

![](../images/wrapper_big_picture_3.png)

---

### Wrapper

.image-10[![](../images/exercise.png)]

<small>[Galaxy Tool XML File](https://docs.galaxyproject.org/en/latest/dev/schema.html)</small>

---

### `command`
#### How to invoke the tool?

```xml
<command><![CDATA[
graphlan.py
--format $format
#if $dpi
--dpi $dpi
#end if
--size $size
"$input_tree"
]]></command>
```

<small>If the script is provided with the wrapper xml</small>
```xml
<command><![CDATA[
python $__tool_directory__/graphlan.py
--format $format
#if $dpi
--dpi $dpi
#end if
--size $size
"$input_tree"
]]></command>
```

---

### `command`
#### How to pass the parameters?

![](../images/graphlan_screenshot.png)

---

### `inputs` > `param` To `command`
#### How to pass the parameters?

```xml
<inputs>
  <param name="dpi" type="..." label="..." optional="True"/>

  <param argument="--size" type="..." value="7" label="..." help="..."/>
</inputs>
```

#### Directly linked to `<command>`

```xml
<command><![CDATA[
graphlan.py
...
#if $dpi
--dpi $dpi
#end if
--size $size
...
]]></command>
```

---

### `inputs` > `param` > `data`

![](../images/input_data.png)

```xml
<param name="..." type="data" format="txt" label="..." help="..." />
```

.my-footer[[List of possible formats](https://github.com/galaxyproject/galaxy/blob/dev/config/datatypes_conf.xml.sample)]

---

### `inputs` > `param` > `integer` | `float`

![](../images/input_integer.png)

```xml
<param name="..." type="integer" value="7" label="..." help="..."/>
```

![](../images/input_integer_range.png)

```xml
<param name="..." type="float" min="0" max="10" value="1" label="..."
help="..."/>
```

---

### `inputs` > `param` > `text`

![](../images/input_text.png)

```xml
<param name="..." type="text" value="" label="..." help="..."/>
```

---

### `inputs` > `param` > `select`

![](../images/input_select.png)

```xml
<param name="..." type="select" label="..." help="...">
  <option value="png" selected="true">PNG</option>
  <option value="pdf">PDF</option>
  <option value="ps">PS</option>
  <option value="eps">EPS</option>
  <option value="svg">SVG</option>
</param>
```

---

### `inputs` > `param` > `select`

![](../images/input_select_checkboxes.png)

```xml
<param name="..." type="select" display="radio" label="..." help="...">
  <option value="min" selected="true">Mininum</option>
  <option value="mean" >Mean</option>
  <option value="max" >Max</option>
  <option value="sum" >Sum</option>
</param>
```

---

### `inputs` > `param` > `select`

![](../images/input_select_multiple.png)

```xml
<param name="..." type="select" multiple="true" label="..." help="...">
  <option value="ld" selected="true">Length distribution</option>
  <option value="gc" selected="true">GC content distribution</option>
</param>
```

---

### `inputs` > `param` > `boolean`

![](../images/input_boolean.png)

```xml
<param name="..." type="boolean" checked="false" truevalue="--log" falsevalue=""
  label="..."  help="..." />
```

---

### `inputs` > `param` > `conditional`

![](../images/input_conditional.png)

```xml
<command><![CDATA[
#if $input_format.format == "fasta":
  phyml_fasta_id_encode_wrapper.sh $input_format.input $output_tree
#end if
#if $input_format.format == "phylip":
  phyml.sh
#end if
]]></command>
<inputs>
  <conditional name="input_format">
    <param type="select" name="format" label="Input format">
      <option value="phylip">Phylip</option>
      <option value="fasta">Fasta</option>
    </param>
    <when value="phylip">
      <param format="phylip" name="input" type="data"  />
    </when>
    <when value="fasta">
      <param format="fasta" name="input" type="data"  />
    </when>
  </conditional>
</inputs>
```

---

### `inputs` > `param` > `repeat`

![](../images/input_repeat.png)

```xml
<command><![CDATA[
#for $i, $s in enumerate( $series )
  rank_of_series=$i
  input_path=${s.input}
  x_colom=${s.xcol}
  y_colom=${s.ycol}
#end for
]]></command>

<inputs>
  <repeat name="series" title="Series">
    <param name="input" type="data" format="tabular" label="Dataset"/>
    <param name="xcol" type="data_column" data_ref="input" label="Column for x axis"/>
    <param name="ycol" type="data_column" data_ref="input" label="Column for y axis"/>
  </repeat>
</inputs>
```

---

### `outputs`
#### What files the tool will produce as output?

![](../images/output.png)

---

### `outputs`

```xml
<outputs>
  <data format="txt" name="tree" label="${tool.name} on ${on_string}: Tree" />
  <data format="txt" name="annotation" label="${tool.name} on ${on_string}:
    Annotation" />
</outputs>
```

---

### `outputs` > `filter`
#### Conditional

```xml
<outputs>
  <data format="png" name="png_output" label="${tool.name} on ${on_string}: PNG">
    <filter>format=="png"</filter>
  </data>
</outputs>
```

---

### `stdio`

```xml
<command detect_errors="aggressive"> ... </command>
```
or if you need more precision
```xml
<stdio>
  <exit_code range=":-2" level="warning" description="Low disk space" />
  <exit_code range="1:" level="fatal"  />
  <regex match="Error:"  level="fatal" />
</stdio>

```

---

### `help`

![](../images/help.png)

```xml
<help><![CDATA[

**What it does**
GraPhlAn is a software tool for producing high-quality circular
representations of taxonomic and phylogenetic trees. GraPhlAn focuses
on concise, integrative, informative, and publication-ready
representations of phylogenetically- and taxonomically-driven
investigation.

For more information, check the `user manual
<https://bitbucket.org/nsegata/graphlan/overview>`_.

]]></help>
```

.my-footer[[reStructuredText markdown format](http://docutils.sourceforge.net/docs/user/rst/quickref.html)]

---

### `citations`

![](../images/citations.png)

```xml
<citations>
  <citation type="doi">10.1093/bioinformatics/bts611</citation>
  <citation type="doi">10.1093/nar/gks1219</citation>
  <citation type="doi">10.1093/nar/gks1005</citation>
  <citation type="doi">10.1093/bioinformatics/btq461</citation>
  <citation type="doi">10.1038/nbt.2198</citation>
</citations>
```

---

.image-25[![](../images/planemo-logo.png)]

#### Building Galaxy Tools

![](../images/planemo_tool_building_lint.png)

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

![](../images/exercise.png)

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### Before: install Seqtk
*No brain needed*

```bash
$ . ~/.venv_planemo/bin activate  # not mandatory
$ planemo conda_init
$ export PATH=~/miniconda2/bin:$PATH
$ conda config --add channels conda-forge
$ conda config --add channels defaults
$ conda config --add channels r
$ conda config --add channels bioconda
$ conda install seqtk
```

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### Before: observe and test Seqtk

```bash
$ seqtk

$ seqtk seq

$ wget https://raw.githubusercontent.com/galaxyproject/galaxy-test-data/master/2.fastq

$ cat 2.fastq

$ seqtk seq -A 2.fastq > 2.fasta

$ cat 2.fasta
```

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### Steps
1. Init the wrapper with `planemo tool_init`
2. Fill the template
3. Test the template with `planemo lint`
<small>Don't worry now about "No tests found"</small>
4. Display the tool with `planemo serve`
5. and try it with a file

???

```bash
# 1.
planemo tool_init
# -> hands_on-tool_integration/seqtk/seqtk_seq_step1.xml
# 2. -> hands_on-tool_integration/seqtk/seqtk_seq_step2.xml
# 3.
planemo lint # seqtk_seq_step2.xml
# 4.
planemo serve # seqtk_seq_step2.xml
```

---




<!-- _____________ FUNCTIONAL TEST _____________ -->

## Functional tests

---

### Functional tests

- Functional testing is a quality assurance (QA) process.
- The tests confort developers and users that the tools can run across different servers/architectures. And that the last modifications didn't break the different features.
- Tools are tested by feeding them input(s) and parameter(s) and examining the output(s) (typically a diff)

---

### Functional tests
#### Good practices

1. Build tests using inputs/parameters/outputs
2. Run the tests --> Failed --> Coffee
3. Develop
4. Run the tests --> Failed  --> Coffee
5. Develop
6. Run the tests --> Succeed --> Beer

---

### `tests`

```xml
<tests>
  <test>
    <param name="input_tree" value="input_tree.txt"/>
    <param name="format" value="png"/>
    <param name="dpi" value="100"/>
    <param name="size" value="7"/>
    <param name="pad" value="2"/>
    <output name="png_output_image" file="png_image.png" />
  </test>
</tests>
```

<small>`input_tree.txt` and `png_image.png` in `test-data` repository</small>

---

### `tests`
#### Comparison Methods

```xml
<output name="out_file1" file="cf_maf2fasta_concat.dat" ftype="fasta"/>
```

```xml
<test>
  <output name="out_file1">
    <assert_contents>
    <has_text text="chr7" />
    <not_has_text text="chr8" />
    <has_text_matching expression="1274\d+53" />
    <has_line_matching expression=".*\s+127489808\s+127494553" />
    <!-- &#009; is XML escape code for tab -->
    <has_line line="chr7&#009;127471195&#009;127489808" />
    <has_n_columns n="3" />
    </assert_contents>
  </output>
</test>
```

.my-footer[[Complete information](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool|tests|test)]

---

### `tests`
#### Comparison Methods on the output itself

```xml
<test>
  <output ...
    md5
    compare
    diff
    sim_size
    lines_diff
    ... />

```

.my-footer[[Complete information](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool|tests|test|output)]

---

### `tests`
#### Comparison Methods within the output

```xml
<test>
  <output>
    <assert_contents>
    <has_text />
    <not_has_text />
    <has_text_matching />
    <has_text />
    <...>
```
.my-footer[[Complete information](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool|tests|test|output|assert_contents)]

---

### Wrapper directory tree

```
graphlan/
├── graphlan.xml
├── graphlan.py
├── test-data/
├── input_tree.txt
├── png_image.png

```

---

.image-25[![](../images/planemo-logo.png)]

#### Test Galaxy Tools

![](../images/planemo_tool_building_test.png)

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

![](../images/exercise.png)

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### The Goal

- Add functional test to the Seqtk wrapper for Seqtk

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### Steps
1. Fill a `<test>` tag using the file you generated previously with `seqtk`
2. Use `planemo test` ... until everything is green

???

```bash
# 1. -> hands_on-tool_integration/seqtk/seqtk_seq_step3.xml
# 2.
planemo test # seqtk_seq_step3.xml
```

---




<!-- _____________ DEPENDENCIES _____________ -->

## Dependencies

---

### Dependencies
#### How Galaxy will deal with dependencies?

![](../images/galaxy_instance.png)

---

### Dependencies
#### Embedded tool

```
graphlan/
├── graphlan.xml
├── graphlan.py

```

```xml
<command><![CDATA[
$__tool_directory__/graphlan.py
...
]]></command>
```

---

### Dependencies
#### External tool

Definition of requirements in `.xml` file

```xml
<requirements>
  <requirement type="package" version="1.66">biopython</requirement>
  <requirement type="package" version="1.0.0">graphlan</requirement>
</requirements>
```

Local installation using ~~tool_dependencies.xml~~

---

### Dependencies
#### External tool

Definition of requirements in `.xml` file

```xml
<requirements>
  <requirement type="package" version="1.66">biopython</requirement>
  <requirement type="package" version="1.0.0">graphlan</requirement>
</requirements>
```

Local installation using `conda` package

---

![](../../shared/images/conda_logo.png)

Package, dependency and environment management

---

.image-50[![](../../shared/images/conda_logo.png)]

- Based on recipes describing how to install the software which are then builded for their distribution
- At installation, no compilation, executables come with their dependencies, libraries...
- Not restricted to Galaxy

---

.image-25[![](../../shared/images/conda_logo.png)]
### Miniconda

![](../images/miniconda_vs_anaconda.png)]

---

.image-25[![](../../shared/images/conda_logo.png)]
### Channels

- Packages are provided through Channels within Continuum.Anaconda. It's the way to contribute.

- Galaxy via Conda will search packages in 5 channels: iuc,bioconda,r,defaults,conda-forge

---

.image-25[![](../../shared/images/conda_logo.png)]
### Quickstart
#### Use Conda outside Planemo

- Install a package


```bash
conda install bioconductor-deseq2
```

- Install some packages within an isolate environment

```bash
$ conda create -n deseq2 bioconductor-deseq2 r-ggplot2
$ conda env list
deseq2        *  ~/miniconda2/envs/deseq2
root             ~/miniconda2
$ source activate bioconductor-deseq2
(deseq2) $ R
```

---

.image-25[![](../../shared/images/conda_logo.png) ]
### Tips and tricks
#### Because you don't want Conda binaries all the time in your $PATH
<small>In fact, Conda will mask some of your binaries like python and so all your Python libraries ...</small>

```bash
$ vi ~/.bashrc
```
```bash
function conda2activate () {
  export PATH=~/miniconda2/bin:$PATH; }
function conda2deactivate () {
  export PATH=$(echo $PATH | sed "s|$HOME/miniconda2/bin:||g"); }
```

---

.image-25[![](../images/planemo-logo.png)]
### Test Galaxy Tools using Conda packages

![](../images/planemo_tool_building.png)

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

![](../images/exercise.png)

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### The Goal
- Use Conda package installation during the functional test of the wrapper for Seqtk

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### Before
1. `conda uninstall seqtk`

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### Steps
1. Fill a `<requirement>` tag to target a `seqtk` conda package
2. Use `planemo conda_install .` to install tools and set an environment
3. Use `planemo test --conda_dependency_resolution`

???

```bash
# 1. -> hands_on-tool_integration/seqtk/seqtk_seq_step4.xml
# 2.
planemo conda_install .
# 3.
planemo test --conda_dependency_resolution # seqtk_seq_step4.xml
```

---

.image-25[![](../../shared/images/conda_logo.png)]
### Write a Conda package

You can implement a new Conda package. This will be necessary if a specific tool or version is not available.

.my-footer[[Documentation about how to build a conda packages from scratch](http://conda.pydata.org/docs/build_tutorials/pkgs2.html) <br/> [Bioconda guidelines](https://bioconda.github.io/guidelines.html)]

---

.image-25[![](../../shared/images/conda_logo.png)]
### Write a Conda package
#### Files

- `meta.yaml`: contains all the metadata of the recipe
- `build.sh`: the Unix script that installs the files

---

.image-25[![](../../shared/images/conda_logo.png)]
### Write a Conda package
#### `meta.yaml` basic information

```yaml
package:
  name: bwa
  version: "0.7.15"
about:
  home: http://bio-bwa.sourceforge.net
  license: MIT
  summary: The BWA read mapper.
build:
  number: 0
  skip: True # [osx]
source:
  fn: v0.7.15.tar.gz
  url: https://github.com/lh3/bwa/archive/v0.7.15.tar.gz
  md5: 54fdee953c5c256d36885a1c5c6b118c
```




---

.image-25[![](../../shared/images/conda_logo.png)]
### Write a Conda package
#### `meta.yaml` requirements

```yaml
requirements:
  build:
    - gcc   # [not osx]
    - llvm  # [osx]
    - zlib
  run:
    - zlib
    - bowtie2
```
<small>
- `build`: requirements needed during the build step (here compilation)
- `run`: requirements needed at runtime

</small>

---

.image-25[![](../../shared/images/conda_logo.png)]
### Write a Conda package
#### `meta.yaml` test

```yaml
test:
  commands:
    - bowtie2 --version
    - bwa 2>&1 | grep 'index sequences in the'
    - '$R -e "library(''xcms'')"'
```
<small>
Those tests are launched at the end of `conda build`
</small>

---

.image-25[![](../../shared/images/conda_logo.png)]
### Write a Conda package
#### `build.sh`

```bash
#!/bin/bash
./configure --prefix=$PREFIX
make
make install
```
```bash
#!/bin/bash
mkdir -p $PREFIX/bin
cp *.py $PREFIX/bin
```
<small>
- $PREFIX will usually contain `bin/` `lib/` `include/` that are filled with the requirements files
- $PREFIX will be loaded in the user environment at runtime: $PATH, LD_LIBRARY_PATH, PYTHONPATH ...

</small>

---

.image-25[![](../../shared/images/conda_logo.png)]
### Write a Conda package
#### skeleton

- `conda skeleton pypi <packagename>`
- `conda skeleton cran <packagename>`
- `bioconductor_skeleton.py <packagename>`
- `conda skeleton cpan <packagename>`

.my-footer[[Building conda packages with conda skeleton](http://conda.pydata.org/docs/build_tutorials/pkgs.html)]

---

.image-25[![](../../shared/images/conda_logo.png)]
### Write a Conda package
#### `conda build` and test the package

```bash
$ conda install conda-build

$ conda build .

```
1. BUILD START: Build/Compile the package
2. BUILD START: Provide a .tar.bz2
3. TEST START: Install the .tar.bz2 previously generated
4. TEST START: Launch the functional tests
5. (Provide the .tar.bz2 path)

---

.image-25[![](../../shared/images/conda_logo.png)]
### Install a Conda package for test
#### `conda install`

To continue your `planemo test`, you may want to use your new package before it's available through an official channel.


1. Get the path of your new package .tar.bz2
2. `conda install ~/miniconda2/conda-bld/linux-64/bwa-0.7.15-0.tar.bz2`

---

.image-50[![](../../shared/images/bioconda_logo.png)]


- A channel dedicated to bioinformatics packages https://anaconda.org/bioconda
- Open to [contribution](https://github.com/bioconda/bioconda-recipes/blob/master/CONTRIBUTING.md)
- GitHub repository: https://github.com/bioconda/bioconda-recipes

---

.image-25[![](../../shared/images/bioconda_logo.png)]

1. Clone `bioconda/bioconda-recipes.git` or a fork
2. Create a new branch "package_new_login"
3. Fill two files `meta.yaml` and `build.sh`
4. Build your new package and test it using `conda build`
5. Push on GitHub
6. Wait for the functional tests to pass on your branch
7. Create a Pull Request on `bioconda/bioconda-recipes.git`
8. Wait for the functional tests to pass on the master branch
9. Enjoy your new Conda package installed from https://anaconda.org/bioconda

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

![](../images/exercise.png)

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### The Goal
Build a conda package for the suite seqtk

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### Steps
1. [RTFM](https://github.com/bioconda/bioconda-recipes/blob/master/GUIDELINES.md#cc)
2. https://github.com/lh3/seqtk/
3. Complete a `meta.yaml` and a `build.sh`
4. Test the build with `conda build .`

???

```bash
# hands_on-conda/seqtk/meta.yaml
# hands_on-conda/seqtk/build.sh
```

---




<!-- _____________ PLANEMO _____________ -->

![](../images/planemo-logo.png)

> Command-line utilities to assist in building and publishing Galaxy tools.

---

.image-25[![](../images/planemo-logo.png)]

![](../images/big_picture.png)

---

.image-25[![](../images/planemo-logo.png)]
#### Checklist (1/3)

- Create a dedicated repository
- Create the main `.xml` file with
```bash
planemo create
```
- Fill the `.xml` file
- Test the wrapper structure with
```bash
planemo lint
```

---

.image-25[![](../images/planemo-logo.png)]
#### Checklist (2/3)

- Check if the dependencies are available
- If not, create some conda packages
- Add functional tests
- Run functional tests
```bash
planemo test
```

---




<!-- _____________ GITHUB + TRAVIS.CI + PLANEMO + CONDA _____________ -->

.image-50[![](../images/github_logo.png)]

.image-50[![](../images/travis_ci_logo.png)]

.image-50[![](../images/planemo-logo.png)]

.image-50[![](../../shared/images/conda_logo.png)]

---

### GitHub side

![](../images/github_travis_github.png)

---

### Travis CI side

![](../images/github_travis_travis.png)

---

.image-50[![](../../shared/images/ansible_logo.png)]

.image-50[![](../../shared/images/conda_logo.png)]

.image-50[![](../../shared/images/docker_logo.png)]

---

### Docker Galaxy Flavors

![](../images/galaxydocker_logo.png)

.my-footer[[Docker Galaxy Stable](https://github.com/bgruening/docker-galaxy-stable)]

---

### Docker Galaxy Flavors
#### Dockerfile

```yaml
# Galaxy - My own flavour
# VERSION       0.1

FROM quay.io/bgruening/galaxy:16.07

MAINTAINER Björn A. Grüning, bjoern.gruening@gmail.com

ENV GALAXY_CONFIG_BRAND deepTools

# Adding the tool definitions to the container
ADD my_tool_list.yml $GALAXY_ROOT/my_tool_list.yml

# Install deepTools
RUN install-tools $GALAXY_ROOT/my_tool_list.yml
```

---

### Docker Galaxy Flavors
#### tool_list.yml

```yaml
api_key: admin
galaxy_instance: http://127.0.0.1:8080/
tools:
- name: fastqc
  owner: devteam
  tool_panel_section_id: cshl_library_information
  tool_shed_url: https://toolshed.g2.bx.psu.edu
  revisions:
    - '8c650f7f76e9'  # v0.62
    - 'd2cf2c0c8a11'  # v0.63
- name: suite_deeptools
  owner: bgruening
  tool_panel_section_label: 'deepTools'
  tool_shed_url: https://toolshed.g2.bx.psu.edu
  revisions:
    - '7f5562625ae2'
```

---

### Docker Galaxy Flavors
#### RUN

```bash
docker run -d -p 8080:80 galaxy-deeptools
```

---




<!-- _____________ ADVANCED _____________ -->

## Advanced features

---

### Other tags

---

### `configfiles`

Like `command` allow cheetah code

- Create a config file
- Create a whole script or a module

---

### `configfiles`

```xml
<command><![CDATA[ mb $script_nexus  ]]></command>

<configfiles>
  <configfile name="script_nexus"><![CDATA[
set autoclose = yes;
execute $input_data;
#if str( $data_type.type ) == "nuc“
  lset nst=$data_type.lset_params.lset_Nst;
#end if
mcmcp ngen=$mcmcp_ngen;
mcmc;
quit
  ]]></configfile>
</configfiles>

```

```bash
set autoclose = yes;
execute dataset_42.dat;
lset nst=2 ;
mcmcp ngen=100000;
mcmc;
quit
```

---

### `command` > Reserved Variables

```xml
<command><![CDATA[
${input_foobar.ext}

--search_pfam "${ filter( lambda x: str( x[0] ) == str( $search.pfam_db ),
  $__app__.tool_data_tables[ 'pfam_databases' ].get_fields() )[0][-1] }"

library $__app__.config.user_library_import_dir/$__user_email__/$inputs.library

#if isinstance( $input1.datatype,
  $__app__.datatypes_registry.get_datatype_by_extension('gff').__class__):
]]></command>
```
.my-footer[[Reserved Variables List](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool|command)]

---

### `macros`

![](../images/macro.png)

.my-footer[[Planemo documents about macros ](http://planemo.readthedocs.io/en/latest/_writing_suites.html#macros)]

---

### `macros`
#### XML elements

macros.xml

```xml
<macros>
  <xml name="requirements">
    <requirements>
      <requirement type="package" version="2.5.0">blast</requirement>
    </requirements>
  </xml>
  <xml name="stdio">
    <stdio>
      <exit_code range="1" level="fatal" />
    </stdio>
  </xml>
</macros>
```

ncbi_blastn_wrapper.xml

```xml
<macros>
  <import>macros.xml</import>
</macros>
<expand macro="requirements"/>
<expand macro="stdio"/>
```

---

### `macros`
#### Tokens

macros.xml

```xml
<macros>
  <token name="@THREADS@">-num_threads "\${GALAXY_SLOTS:-8}"</token>
</macros>
```

ncbi_blastn_wrapper.xml

```xml
<command>
  blastn -query "$query" @THREADS@ [...]
</command>
```

---

### `macros`
#### `<yield/>`

macros.xml

```xml
<macros>
  <xml name="requirements">
    <requirements>
      <requirement type="package" version="2.2.0">trinity</requirement>
      <yield/>
    </requirements>
  </xml>
</macros>
```

trinity.xml

```xml
<expand macro="requirements">
  <requirement type="package" version="1.1.2">bowtie</requirement>
  <requirement type="set_environment">TRINITY_MEM_OPTIONS</requirement>
</expand>
```

---

### Dataset collection

To group related datasets, and treat them as a single unit

- `list`: a simple list of files
- `paired`: a pair of files, `forward` and `reverse` for NGS
- composite: e.g. `list:paired` for a list of dataset pairs

---

### Dataset collection > input

Accept both datasets and collections:

```xml
<param type="data" multiple="true" name="inputs" label="Input BAM(s)"
  format="bam" />
```

Accept only collections:

```xml
<param type="data_collection" collection_type="list|paired|list:paired|..."
  name="inputs" label="Input BAM(s)" format="bam" />
```

In the command:

```xml
<command><![CDATA[
...
#for $input in $inputs
--input "$input"
--sample_name "$input.element_identifier"
#end for
]]></command>
```

---

### Dataset collection > output

A single paired collection:

```xml
<collection name="paired_output" type="paired" label="Split Pair">
  <data name="forward" format="txt" />
  <data name="reverse" format_source="input1" from_work_dir="reverse.txt" />
</collection>
```

Unknown number of files:

```xml
<collection name="output" type="list" label="Unknown number of files">
  <discover_datasets pattern="__name_and_ext__" directory="outputs" />
</collection>
```


`__name_and_ext__`: a predefined regexp,

catches the dataset identifier + the format

---

### Some Config

---

### `job_conf.xml`

![](../images/job_conf.xml_1.png)

```bash
blastn -query foo_bar -num_threads 8
```

---

### `job_conf.xml`

![](../images/job_conf.xml_2.png)

```bash
blastn -query foo_bar -num_threads 4
```

---

### `tool_data_table_conf.xml`

![](../images/tool_data_table_conf.xml.png)

---

### Datatypes

---

### Datatypes

- Every Galaxy dataset is associated with a datatype
- Datatype can be detected or user specified
- Gain of usability

.my-footer[[Documentation: Adding Datatypes](https://wiki.galaxyproject.org/Admin/Datatypes/Adding%20Datatypes)]

---




<!-- _____________ CONCLUSION _____________ -->

## Conclusion

---

### IUC: Intergalactic Utilities Commission

- Lobbyist :)
- Host orphan tools: [GitHub](https://github.com/galaxyproject/tools-iuc)
- [IUC-standards](http://galaxy-iuc-standards.readthedocs.io/)

---

### <i class="fa fa-exclamation-circle" aria-hidden="true"></i> Key points

- Galaxy Tool Syntax
- Use Planemo
- Use Conda
- Use GitHub
- Use Travis
- No more excuse to develop crappy Galaxy tools
