---
layout: tutorial_slides
topic_name: visualization
tutorial_name: rna-seq-viz-with-cummerbund
logo: "GTN"
---

# Why visualization?

???
Data alone does not bring any information: to carry information, data needs to be contextualized. Transcriptomic data is no exception, therefore to organize the growing body of knowledge pertaining RNA-Seq experiments and infer valuable insights, data needs to be organized, annotated, and ultimately visualized.

---

### Where is my data coming from?

![](../../images/rna-seq-experiment.jpg)

<small>[*Wang et al, Nat. Rev. Genet., 2009*](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2949280/)</small>

???
RNAs are first converted into a cDNA fragments library through RNA fragmentation. Sequencing adaptors (in blue) are then added to each cDNA fragment, and the cDNA sequence is obtained via NGS sequencing. The obtained reads are subsequently aligned against a reference genome or transcriptome, and classified in-silico as exonic reads, junction reads, and poly-A reads. These three types are then used to outline am expression profile for each gene.

---

### Second slide

![](../images/RNA_seq_zang2016.png)

<small>[*Zang and Mortazavi, Nature, 2012*](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4138050/)</small>

---

### A Third slide

Some text

???

Slide notes
-> Pressing **P** will toggle presenter mode.


---

# Part 2

---

### An other slide

With text and a resized image

.image-25[![](../images/ecker_2012.jpg)]
