RAD-Seq data analysis
======================

RAD-Sequencing is a reduced representation library method used to obtain genomic information on a whole genome without sequencing each nucleotides. Here, we propose some materials to learn how to analyze RAD-Seq data.

# Slides

Several deck of slides are available for this topic:

- [General introduction about RAD seq data analysis](http://bgruening.github.io/training-material/RAD-Seq/slides/)
- Slide deck related to the tutorials:
    - [Population genomics data analysis](http://bgruening.github.io/training-material/RAD-Seq/slides/pop_genomics.html)
	- [Genetic map data analysis](http://bgruening.github.io/training-material/RAD-Seq/slides/genetic_map.html)

# Tutorials

A tutorial with hands-on is available for this topic:

- [Reference-based RAD-Seq data analysis](tutorials/ref_based_rad.md)
- [de-novo RAD-Seq data analysis](tutorials/de_novo_rad.md)

## Input datasets

The input datasets for the tutorials are available on Zenodo:
- [population gneomics tutorials](http://doi.org/10.5281/zenodo.218574).

## Galaxy instance

For these tutorials, you can use the [dedicated Docker image](docker/README.md):

```
docker run -d -p 8080:80 bgruening/galaxy-RAD-Seq-training
```

It will launch a flavored Galaxy instance available on
[http://localhost:8080](http://localhost:8080).

# References

## Papers

**Amores, A., et al. 2011:** [Genome evolution and meiotic maps by massively parallel DNA sequencing] (https://www.ncbi.nlm.nih.gov/pubmed/21828280)

**Catchen, J. M. et al. 2011:** [Stacks: building and genotyping loci de novo from short-read sequences] (http://www.g3journal.org/content/1/3/171.full)

**Davey, J. W., et al. 2011:** [Genome-wide genetic marker discovery and genotyping using next-generation sequencing] (https://www.ncbi.nlm.nih.gov/pubmed/21681211)

**Etter, P. D., et al. 2011:** [SNP Discovery and Genotyping for Evolutionary Genetics using RAD sequencing] (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3658458/)

**Ekblom, R., and J. Galindo. 2010:** [Applications of next generation sequencing in molecular ecology of non-model organisms] (https://www.ncbi.nlm.nih.gov/pubmed/21139633)

**Hohenlohe, P. A. et al. 2010:** [Population genomics of parallel adaptation in threespine stickleback using sequenced RAD tags] (http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1000862)

**Gompert, Z., and C. A. Buerkle. 2011a:** [A hierarchical Bayesian model for next-generation population genomics] (http://www.genetics.org/content/187/3/903)

**Gompert, Z., and C. A. Buerkle. 2011b:** [Bayesian estimation of genomic clines] (http://onlinelibrary.wiley.com/doi/10.1111/j.1365-294X.2011.05074.x/abstract)

**Lynch, M. 2009:** [Estimation of allele frequencies from high-coverage genome-sequencing projects] (http://www.genetics.org/content/182/1/295)

**Nielsen, R., et al. 2005:** [Genomic scans for selective sweeps using SNP data] (http://genome.cshlp.org/content/15/11/1566.abstract)

**Hohenlohe, P. A., et al. 2010:** [Using population genomics to detect selection in natural populations: Key concepts and methodological considerations] (https://www.ncbi.nlm.nih.gov/pubmed/21218185)

**Stapley, J., et al. 2010:** [Adaptation genomics: the next generation] (http://www.cell.com/trends/ecology-evolution/abstract/S0169-5347(10)00216-8)

**Luikart, G.,et al. 2003:** [The power and promise of population genomics: from genotyping to genome typing] (http://www.nature.com/nrg/journal/v4/n12/full/nrg1226.html)

**Nielsen, R., et al. 2011:** [Genotype and SNP calling from next-generation sequencing data] (http://www.nature.com/nrg/journal/v12/n6/full/nrg2986.html)

### Genetic mapping using RRL and RAD sequencing

**Altshuler, D., et al. 2000:** [An SNP map of the human genome generated by reduced representation shotgun sequencing] (https://www.ncbi.nlm.nih.gov/pubmed/11029002)

**Baxter, S. W., et al. 2011:** [Linkage mapping and comparative genomics using next-generation RAD sequencing of a non-model organism] (http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0019315)

**Chutimanitsakun, Y., et al. 2011:** [Construction and application for QTL analysis of a Restriction Site Associated DNA (RAD) linkage map in barley] (http://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-12-4)

**Gore, M. A., et al. 2009:** [A first-generation haplotype map of maize] (http://science.sciencemag.org/content/326/5956/1115)

### RAD-seq genotyping methodology

**Baird, N. A., et al. 2008:** [Rapid SNP discovery and genetic mapping using sequenced RAD markers] (http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0003376)

**Emerson, K. J., et al. 2010:** [Resolving postglacial phylogeography using high-throughput sequencing] (http://www.pnas.org/content/107/37/16196)

**Etter, P. D., et al. 2011:** [Local De Novo Assembly of RAD Paired-End Contigs Using Short Sequencing Reads] (http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0018561)

**Hohenlohe, P. A., et al. 2011:** [Next-generation RAD sequencing identifies thousands of SNPs for assessing hybridization between rainbow and westslope cutthroat trout] (https://www.ncbi.nlm.nih.gov/pubmed/21429168)

**Miller, M. R., et al. 2007:** [Rapid and cost-effective polymorphism identification and genotyping using restriction site associated DNA (RAD) markers] (https://www.ncbi.nlm.nih.gov/pubmed/17189378)

**Willing, E. M., et al. 2011:** [Paired-end RAD-seq for de novo assembly and marker design without available reference] (https://www.ncbi.nlm.nih.gov/pubmed/21712251)

### Other reduced representation library (RRL) methodologies

**Andolfatto, P., et al. 2011:** [Multiplexed shotgun genotyping for rapid and efficient genetic mapping] (https://www.ncbi.nlm.nih.gov/pubmed/21233398)

**Elshire, R. J., et al. 2011:** [A Robust, Simple Genotyping-by-Sequencing (GBS) Approach for High Diversity Species] (http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0019379)

**Rigola, D., et al. 2009:** [High-Throughput Detection of Induced Mutations and Natural Variation Using KeyPointâ„¢ Technology (http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0004761)

**van Orsouw, N. J., et al. 2007:** [Complexity reduction of polymorphic sequences (CRoPS): a novel approach for large-scale polymorphism discovery in complex genomes] (http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0001172)

**van Tassell, C. P., et al. 2008:** [SNP discovery and allele frequency estimation by deep sequencing of reduced representation libraries] (https://www.ncbi.nlm.nih.gov/pubmed/18297082)


## Websites

### Stacks infos and tutorials

http://creskolab.uoregon.edu/stacks/

http://evomics.org/learning/genomics/stacks/

### Great site for information on next gen sequencing

http://seqanswers.com/


# Contributors

This material is maintained by:

- Yvan Le Bras

For any question related to this topic and the content, you can contact them.

The following individuals have contributed to this training material:

- Yvan Le Bras

