<!DOCTYPE html>
<html>
    <head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <title>Galaxy Training!</title>
        <link media="all" href="https://netdna.bootstrapcdn.com/bootstrap/3.2.0/css/bootstrap.min.css" rel="stylesheet">
        <link rel="stylesheet" href="https://galaxyproject.github.io/training-material//assets/css/main.css">
        <link rel="stylesheet" href="https://galaxyproject.github.io/training-material//assets/css/font-awesome.css">
        <link rel="stylesheet" href="https://galaxyproject.github.io/training-material//assets/css/syntax_highlighting.css">
    </head>
    <body>
        <div class="wrapper">
            <header>
  <h1><a href="https://galaxyproject.github.io/training-material/">Galaxy Training!</a></h1>
  <p>Collection of tutorials developed and maintained by the worldwide Galaxy community.</p>
  <a href="https://gitter.im/Galaxy-Training-Network/Lobby?utm_source=badge&amp;utm_medium=badge&amp;utm_campaign=pr-badge&amp;utm_content=badge">
    <img src="https://camo.githubusercontent.com/66dbecf63f6c9c68127a18afaca4790a7eb6f1c1/68747470733a2f2f6261646765732e6769747465722e696d2f47616c6178792d547261696e696e672d4e6574776f726b2f747261696e696e672d6d6174657269616c2e737667" alt="Gitter" data-canonical-src="https://badges.gitter.im/Galaxy-Training-Network/training-material.svg" style="max-width:100%;">
  </a>
</header>



<section>
    <h1>ChIP-Seq data analysis</h1>

    <p>ChIP-sequencing is a method used to analyze protein interactions with DNA.</p>

    
        <a name="requirements"></a>
        <h2>Requirements</h2>

        <p>Before diving into this topic, we recommend you to have a look at:</p>

        <ul>
            
                <li>
                    
                    <a href="https://galaxyproject.github.io/training-material//topics/introduction/">Galaxy introduction</a>
                    
                </li>
            
                <li>
                    
                    <a href="https://galaxyproject.github.io/training-material//topics/sequence-analysis/">Quality control</a>
                    
                </li>
            
                <li>
                    
                    <a href="https://galaxyproject.github.io/training-material//topics/sequence-analysis/">Mapping</a>
                    
                </li>
            
        </ul>
    

    <a name="material"></a>
    <h2>Material</h2>

    <table class="table table-striped">
        <tr>
              <td class="bold">Lesson</td>
              <td class="bold">Hands-on</td>
              <td class="bold">Slides</td>

              
                  <td class="bold">Input dataset</td>
                  <td class="bold">Galaxy tour</td>
              
        </tr>

        
            
                <tr>
                    <td>Introduction</td>

                    
                        <td></td>
                        <td>
                            
                            <a href="https://galaxyproject.github.io/training-material///topics/chip-seq/slides"><img class="emoji" title=":mortar_board:" alt=":mortar_board:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f393.png" height="20" width="20" align="absmiddle"></a>
                            
                        </td>
                        
                            <td></td>
                            <td></td>
                        

                    
                </tr>
            
        
            
                <tr>
                    <td>Identification of the binding sites of the T-cell acute lymphocytic leukemia protein 1 (TAL1)</td>

                    
                        <td>
                            
                            <a href="https://galaxyproject.github.io/training-material//topics/chip-seq/tutorials/tal1-binding-site-identification/tutorial.html"><img class="emoji" title=":book:" alt=":book:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f4d6.png" height="20" width="20" align="absmiddle"></a>
                            
                        </td>

                        <td>
                            
                        </td>

                        
                            <td>
                                
                                <a href="https://doi.org/10.5281/zenodo.197100"><img class="emoji" title=":open_file_folder:" alt=":open_file_folder:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f4c2.png" height="20" width="20" align="absmiddle"></a>
                                
                            </td>

                            <td>
                                
                            </td>
                        
                    
                </tr>
            
        
            
        
    </table>

    
        <a name="docker_image"></a>
        <h2>Galaxy instance</h2>

        <p>For these tutorials, you can use the dedicated Docker image:</p>

        <pre><code>
        docker run -d -p 8080:80 bgruening/galaxy-chip-seq-training
        </code></pre>

        <p>It will launch a flavored Galaxy instance available on <a href="http://localhost:8080">http://localhost:8080</a>.</p>
    

    

    <a name="contributors"></a>
    <h2>Contributors</h2>

    <p>This material is maintained by:</p>

    <ul>
        
            <li>
            
                <a href="https://github.com/malloryfreeberg">Mallory Freeberg</a>
            
            
            </li>
        
            <li>
            
                <a href="https://github.com/MoHeydarian">Mo Heydarian</a>
            
            
            </li>
        
    </ul>

    <p><em>For any question related to this topic and the content, you can contact them.</em></p>

    <p>The following persons have contributed to this training material:</p>

    <ul>
        
            <li>
            
                <a href="https://github.com/malloryfreeberg">Mallory Freeberg</a>
            
            
            </li>
        
            <li>
            
                <a href="https://github.com/MoHeydarian">Mo Heydarian</a>
            
            
            </li>
        
            <li>
            
                <a href="https://github.com/friedue">Friederike Dündar</a>
            
            
            </li>
        
            <li>
            
                <a href="https://github.com/bebatut">Bérénice Batut</a>
            
            
                (<a href="mailto:berenice.batut@gmail.com">berenice.batut@gmail.com</a>)
            
            </li>
        
    </ul>

    
        <a name="references"></a>
        <h2>References</h2>

        
            <p>
            <span class="bold">Stephen G Landt et al:</span> <a href="http://genome.cshlp.org/content/22/9/1813.long">ChIP-seq guidelines and practices of the ENCODE and modENCODE consortia</a>

            
            <br><cite>A very useful encyclopedic paper with many details about the tools the (mod)ENCODE consortia use and also contains a long section about antibody validation</cite>
            
            </p>
        
            <p>
            <span class="bold">Gabriel E Zentner and Steven Henikoff:</span> <a href="http://genomebiology.biomedcentral.com/articles/10.1186/gb-2012-13-10-250">Surveying the epigenomic landscape, one base at a time</a>

            
            <br><cite>Overview of popular sequencing techniques with very nice descriptions of DNase-seq, MNase-seq, FAIRE-seq</cite>
            
            </p>
        
            <p>
            <span class="bold">Benjamin L Kidder et al:</span> <a href="http://www.nature.com/ni/journal/v12/n10/abs/ni.2117.html">ChIP-Seq: technical considerations for obtaining high-quality data</a>

            
            <br><cite>Nice, readable introduction into all aspects of ChIP-seq experiments (from antibodies to cell numbers to replicates to data analysis)</cite>
            
            </p>
        
            <p>
            <span class="bold">Marion Leleu et al:</span> <a href="http://bfg.oxfordjournals.org/content/9/5-6/466">Processing and analyzing ChIP-seq data</a>

            
            <br><cite>Fairly detailed review of key concepts of ChIP-seq data processing (less detailed on analysis)</cite>
            
            </p>
        
            <p>
            <span class="bold">Peter J. Park:</span> <a href="http://www.nature.com/nrg/journal/v10/n10/full/nrg2641.html">ChIP-seq: Advantages and challenges of a maturing technology</a>

            
            </p>
        
            <p>
            <span class="bold">Peter V Kharchenko et al:</span> <a href="http://www.nature.com/nbt/journal/v26/n12/full/nbt.1508.html">Design and analysis of ChIP-seq experiments for DNA-binding proteins</a>

            
            </p>
        
            <p>
            <span class="bold">Edison T Liu et al:</span> <a href="http://bmcbiol.biomedcentral.com/articles/10.1186/1741-7007-8-56">Q&amp;A: ChIP-seq technologies and the study of gene regulation</a>

            
            <br><cite>Short overview of several (typical) issues of ChIP-seq analysis</cite>
            
            </p>
        
            <p>
            <span class="bold">Thomas S. Carroll et al:</span> <a href="http://journal.frontiersin.org/article/10.3389/fgene.2014.00075/full">Impact of artifact removal on ChIP quality metrics in ChIP-seq and ChIP-exo data</a>

            
            </p>
        
            <p>
            <span class="bold">Shirley Pepke et al:</span> <a href="http://www.nature.com/nmeth/journal/v6/n11s/full/nmeth.1371.html">Computation for ChIP-seq and RNA-seq studies</a>

            
            <br><cite>First comparison of peak callers, focuses on the explanation of basic principles of ChIP-seq data processing and general workflows of peak calling algorithms</cite>
            
            </p>
        
            <p>
            <span class="bold">Elizabeth G. Wilbanks &amp; Marc T. Facciotti:</span> <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0011471">Evaluation of Algorithm Performance in ChIP-Seq Peak Detection</a>

            
            <br><cite>Another comparison of peak callers - focuses more on the evaluation of the peak callers performances than Shirley Pepke et al.</cite>
            
            </p>
        
            <p>
            <span class="bold">Mariann Micsinai et al:</span> <a href="http://nar.oxfordjournals.org/content/40/9/e70.full">Picking ChIP-seq peak detectors for analyzing chromatin modification experiments</a>

            
            <br><cite>How to choose the best peak caller for your data set - their finding: default parameters, surprisingly, yield the most reproducible results regardless of the data set type</cite>
            
            </p>
        
            <p>
            <span class="bold">Jianxing Fen et al:</span> <a href="http://www.nature.com/nprot/journal/v7/n9/abs/nprot.2012.101.html">Identifying ChIP-seq enrichment using MACS</a>

            
            <br><cite>How to use MACS</cite>
            
            </p>
        
            <p>
            <span class="bold">Yong Zhang et al:</span> <a href="http://genomebiology.biomedcentral.com/articles/10.1186/gb-2008-9-9-r137">Model-based Analysis of ChIP-Seq (MACS)</a>

            
            <br><cite>Original publication of MACS</cite>
            
            </p>
        
            <p>
            <span class="bold">Modan K Das &amp; Ho-Kwok Dai:</span> <a href="http://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-8-S7-S21">A survey of DNA motif finding algorithms</a>

            
            <br><cite>Review of motif analysis tools</cite>
            
            </p>
        
            <p>
            <span class="bold">Philip Machanick and Timothy L. Bailey:</span> <a href="http://bioinformatics.oxfordjournals.org/content/27/12/1696.short">MEME-ChIP: motif analysis of large DNA datasets</a>

            
            <br><cite>MEME-ChIP-paper</cite>
            
            </p>
        
            <p>
            <span class="bold">Timothy L. Bailey and Philip Machanick:</span> <a href="http://nar.oxfordjournals.org/content/40/17/e128">Inferring direct DNA binding from ChIP-seq</a>

            
            <br><cite>Centrimo: position-specific motif analysis, especially useful for ChIP-seq data</cite>
            
            </p>
        
            <p>
            <span class="bold">Morgane Thomas-Chollier et al:</span> <a href="http://www.nature.com/nprot/journal/v6/n12/abs/nprot.2011.409.html">Transcription factor binding predictions using TRAP for the analysis of ChIP-seq data and regulatory SNPs</a>

            
            <br><cite>How to use TRAP</cite>
            
            </p>
        
            <p>
            <span class="bold">Helge G. Roider et al:</span> <a href="http://bioinformatics.oxfordjournals.org/content/23/2/134.short">Predicting transcription factor affinities to DNA from a biophysical model</a>

            
            <br><cite>Theoretical background of TRAP</cite>
            
            </p>
        
    
</section>

<footer>
  <p>
    The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a> helps researchers find online training material, and trainers in their geographic area, and helps advertise training events as well.
  </p>
  <img src="https://galaxyproject.github.io/training-material///assets/images/GTN.png" alt="Galaxy Training Network logo" style="max-width: 222px;">
</footer>


        </div>
    </body>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
</html>
